<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement" />


<title>6. Simple linear regression</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>


<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PSLS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01-intro.html">1. Introduction</a>
    </li>
    <li>
      <a href="02-concepts.html">2. Concepts</a>
    </li>
    <li>
      <a href="03-experimentalDesign.html">3. Experimental Design I</a>
    </li>
    <li>
      <a href="04-dataExploration.html">4. Data Exploration</a>
    </li>
    <li>
      <a href="05-statisticalInference.html">5.1 Statistical Inference 1</a>
    </li>
    <li>
      <a href="05-statisticalInference-twosampleT.html">5.2 Statistical Inference 2</a>
    </li>
    <li>
      <a href="06-linearRegression.html">6. Linear Regression</a>
    </li>
    <li>
      <a href="07-Anova.html">7. Analysis of Variance</a>
    </li>
    <li>
      <a href="08-MultipleRegression.html">8. Multiple Regression</a>
    </li>
    <li>
      <a href="08-MultipleRegression-SpecificDesigns.html">8.2 Multiple Regression: Factorial Designs</a>
    </li>
    <li>
      <a href="08-experimentalDesignII.html">8.3. Experimental Design II</a>
    </li>
    <li>
      <a href="08-experimentalDesignIII.html">8.4. Experimental Design II</a>
    </li>
    <li>
      <a href="09-NonparametericStatistics-WilcoxonMannWithney.html">9.1. Nonparametric Statistics 1</a>
    </li>
    <li>
      <a href="09-NonparametericStatistics-KruskalWallis.html">9.2. Nonparametric Statistics 2</a>
    </li>
    <li>
      <a href="10-categoricalDataAnalysis.html">10. Categorical Data Analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    Labs
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="04_exercises.html">4. Data Exploration</a>
    </li>
    <li>
      <a href="05_exercises.html">5. Statistical inference</a>
    </li>
    <li>
      <a href="06_exercises.html">6. Simple linear regression</a>
    </li>
    <li>
      <a href="07_exercises.html">7. ANOVA</a>
    </li>
    <li>
      <a href="08_exercises.html">8. Multiple linear regression</a>
    </li>
    <li>
      <a href="09_exercises.html">9. Non-parametric statistics</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/statOmics/PSLS">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="http://statomics.github.io/">statOmics</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">6. Simple linear regression</h1>
<h4 class="author">Lieven Clement</h4>
<h4 class="date">statOmics, Ghent University (<a href="https://statomics.github.io" class="uri">https://statomics.github.io</a>)</h4>

</div>


<p><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a></p>
<div id="breast-cancer-dataset" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Breast cancer dataset</h1>
<ul>
<li><p>Subset of study <a href="https://doi.org/10.1093/jnci/djj052" class="uri">https://doi.org/10.1093/jnci/djj052</a></p></li>
<li><p>32 breast cancer patients with estrogen recepter positieve tumor that had tamoxifen chemotherapy. Variabels:</p>
<ul>
<li>grade: histological grade of tumor (grade 1 vs 3),</li>
<li>node: lymph node status (0: not affected, 1: lymph nodes affected and removed),</li>
<li>size: tumor size in cm,</li>
<li>ESR1 and S100A8 gene expression in tumor biopsy (microarray technology)</li>
</ul></li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>brca <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;https://raw.githubusercontent.com/GTPB/PSLS20/master/data/breastcancer.csv&quot;</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>brca</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["sample_name"],"name":[1],"type":["chr"],"align":["left"]},{"label":["filename"],"name":[2],"type":["chr"],"align":["left"]},{"label":["treatment"],"name":[3],"type":["chr"],"align":["left"]},{"label":["er"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["grade"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["node"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["size"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["age"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["ESR1"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["S100A8"],"name":[10],"type":["dbl"],"align":["right"]}],"data":[{"1":"OXFT_209","2":"gsm65344.cel.gz","3":"tamoxifen","4":"1","5":"3","6":"1","7":"2.5","8":"66","9":"1939.1990","10":"207.19682"},{"1":"OXFT_1769","2":"gsm65345.cel.gz","3":"tamoxifen","4":"1","5":"1","6":"1","7":"3.5","8":"86","9":"2751.9521","10":"36.98611"},{"1":"OXFT_2093","2":"gsm65347.cel.gz","3":"tamoxifen","4":"1","5":"1","6":"1","7":"2.2","8":"74","9":"379.1951","10":"2364.18306"},{"1":"OXFT_1770","2":"gsm65348.cel.gz","3":"tamoxifen","4":"1","5":"1","6":"1","7":"1.7","8":"69","9":"2531.7473","10":"23.61504"},{"1":"OXFT_1342","2":"gsm65350.cel.gz","3":"tamoxifen","4":"1","5":"3","6":"0","7":"2.5","8":"62","9":"141.0508","10":"3218.74109"},{"1":"OXFT_2338","2":"gsm65352.cel.gz","3":"tamoxifen","4":"1","5":"3","6":"1","7":"1.4","8":"63","9":"1495.4213","10":"107.56868"},{"1":"OXFT_2341","2":"gsm65353.cel.gz","3":"tamoxifen","4":"1","5":"1","6":"1","7":"3.3","8":"76","9":"3405.7294","10":"13.99662"},{"1":"OXFT_1902","2":"gsm65354.cel.gz","3":"tamoxifen","4":"1","5":"3","6":"0","7":"2.4","8":"61","9":"2812.8034","10":"68.38262"},{"1":"OXFT_1982","2":"gsm65355.cel.gz","3":"tamoxifen","4":"1","5":"1","6":"0","7":"1.7","8":"62","9":"949.9327","10":"74.18828"},{"1":"OXFT_5210","2":"gsm65356.cel.gz","3":"tamoxifen","4":"1","5":"3","6":"0","7":"3.5","8":"65","9":"1052.9142","10":"182.32133"},{"1":"OXFT_2027","2":"gsm65357.cel.gz","3":"tamoxifen","4":"1","5":"3","6":"0","7":"1.4","8":"63","9":"1838.7705","10":"27.51279"},{"1":"OXFT_1133","2":"gsm65358.cel.gz","3":"tamoxifen","4":"1","5":"1","6":"0","7":"2.0","8":"70","9":"3707.8010","10":"26.70991"},{"1":"OXFT_1441","2":"gsm65359.cel.gz","3":"tamoxifen","4":"1","5":"3","6":"0","7":"3.2","8":"78","9":"2240.4095","10":"26.28934"},{"1":"OXFT_1432","2":"gsm65360.cel.gz","3":"tamoxifen","4":"1","5":"1","6":"0","7":"2.0","8":"71","9":"2698.5507","10":"56.30524"},{"1":"OXFT_1125","2":"gsm65361.cel.gz","3":"tamoxifen","4":"1","5":"1","6":"0","7":"1.6","8":"68","9":"2459.7793","10":"14.17095"},{"1":"OXFT_1070","2":"gsm65362.cel.gz","3":"tamoxifen","4":"1","5":"1","6":"0","7":"0.2","8":"45","9":"689.2472","10":"282.68343"},{"1":"OXFT_700","2":"gsm65363.cel.gz","3":"tamoxifen","4":"1","5":"1","6":"0","7":"3.5","8":"57","9":"2625.8917","10":"18.67950"},{"1":"OXFT_742","2":"gsm65364.cel.gz","3":"tamoxifen","4":"1","5":"3","6":"1","7":"1.7","8":"67","9":"1665.5595","10":"18.11686"},{"1":"OXFT_738","2":"gsm65365.cel.gz","3":"tamoxifen","4":"1","5":"1","6":"1","7":"1.8","8":"74","9":"3336.0416","10":"26.00373"},{"1":"OXFT_669","2":"gsm65366.cel.gz","3":"tamoxifen","4":"1","5":"3","6":"0","7":"4.0","8":"60","9":"3057.5784","10":"22.59829"},{"1":"OXFT_3597","2":"gsm65367.cel.gz","3":"tamoxifen","4":"1","5":"3","6":"0","7":"1.1","8":"60","9":"164.2995","10":"6225.02863"},{"1":"OXFT_638","2":"gsm65368.cel.gz","3":"tamoxifen","4":"1","5":"1","6":"1","7":"2.3","8":"63","9":"2909.6771","10":"78.53095"},{"1":"OXFT_2188","2":"gsm65369.cel.gz","3":"tamoxifen","4":"1","5":"3","6":"0","7":"3.3","8":"69","9":"1925.8788","10":"82.15699"},{"1":"OXFT_2069","2":"gsm65370.cel.gz","3":"tamoxifen","4":"1","5":"3","6":"1","7":"2.3","8":"66","9":"1864.3941","10":"65.50852"},{"1":"OXFT_596","2":"gsm65371.cel.gz","3":"tamoxifen","4":"1","5":"3","6":"0","7":"3.0","8":"66","9":"3037.2311","10":"21.68372"},{"1":"OXFT_619","2":"gsm65372.cel.gz","3":"tamoxifen","4":"1","5":"3","6":"0","7":"1.9","8":"49","9":"1449.9249","10":"27.11254"},{"1":"OXFT_680","2":"gsm65373.cel.gz","3":"tamoxifen","4":"1","5":"1","6":"1","7":"4.5","8":"61","9":"3967.2038","10":"26.73890"},{"1":"OXFT_443","2":"gsm65374.cel.gz","3":"tamoxifen","4":"1","5":"3","6":"0","7":"2.1","8":"45","9":"396.0846","10":"219.23261"},{"1":"OXFT_1579","2":"gsm65376.cel.gz","3":"tamoxifen","4":"1","5":"1","6":"0","7":"2.0","8":"40","9":"976.9015","10":"116.66995"},{"1":"OXFT_4904","2":"gsm65377.cel.gz","3":"tamoxifen","4":"1","5":"1","6":"1","7":"5.0","8":"69","9":"2036.9399","10":"102.14641"},{"1":"OXFT_511","2":"gsm65378.cel.gz","3":"tamoxifen","4":"1","5":"3","6":"1","7":"5.5","8":"64","9":"1004.3520","10":"294.16364"},{"1":"OXFT_736","2":"gsm65379.cel.gz","3":"tamoxifen","4":"1","5":"1","6":"0","7":"1.4","8":"48","9":"2163.5382","10":"45.28436"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<ul>
<li>For didactical reasons we first remove 3 outliers in the S100A8 expression data.</li>
<li>Later in the lecture we will show how to properly deal with all data.</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>brca <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="st">&quot;&quot;</span>,<span class="at">y=</span>S100A8)) <span class="sc">+</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_boxplot</span>()</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-2-1.png" width="70%" style="display: block; margin: auto;" /></p>
<hr />
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GGally)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>brcaSubset<span class="ot">&lt;-</span>brca <span class="sc">%&gt;%</span> <span class="fu">filter</span>(S100A8<span class="sc">&lt;</span><span class="dv">2000</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>brcaSubset[,<span class="sc">-</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)] <span class="sc">%&gt;%</span> <span class="fu">ggpairs</span>()</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<div id="association-between-esr1-and-s100a8-expressie" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Association between ESR1 and S100A8 expressie</h2>
<ul>
<li><p>ESR1 in <span class="math inline">\(\pm\)</span> 75% of breast cancer tumors.</p>
<ul>
<li>Expression of ER gene positive for treatment: tumor responds to hormone therapy</li>
<li>Tamoxifen interacts with ER and modulates gene expression.</li>
</ul></li>
<li><p>Proteins of S100 family often dysregulated in cancer</p>
<ul>
<li>S100A8 expressie represses immune systeem in tumor en creates an environment of inflamation that promotes tumor growth.</li>
</ul></li>
<li><p>Assess association between ESR1 and S100A8 expression.</p></li>
</ul>
<ol style="list-style-type: decimal">
<li>pipe dataset to ggplot</li>
<li>select data <code>ggplot(aes(x=ESR1,y=S100A8))</code></li>
<li>add points <code>geom_point()</code></li>
<li>add smooth line <code>geom_smooth()</code></li>
</ol>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>brcaSubset <span class="sc">%&gt;%</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ESR1,<span class="at">y=</span>S100A8)) <span class="sc">+</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>()</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="lineair-regression" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Lineair Regression</h1>
<ul>
<li><p>Statistical method to assess association between two variables <span class="math inline">\((X_i, Y_i)\)</span>, measured on each subject <span class="math inline">\(i = 1, ..., n\)</span>.</p></li>
<li><p>Gene expression example</p>
<ul>
<li>Response Y : S100A8 expression</li>
<li>Predictor X: ESR1 expression</li>
</ul></li>
</ul>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>brcaSubset <span class="sc">%&gt;%</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ESR1,<span class="at">y=</span>S100A8)) <span class="sc">+</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">se=</span><span class="cn">FALSE</span>,<span class="at">col=</span><span class="st">&quot;grey&quot;</span>) <span class="sc">+</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">&quot;lm&quot;</span>,<span class="at">se=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="model" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Model</h2>
<ul>
<li>For fixed <span class="math inline">\(X\)</span>, <span class="math inline">\(Y\)</span> does not necessarly has the same value</li>
</ul>
<p><span class="math display">\[\text{observation = signal + noise}\]</span></p>
<p><span class="math display">\[Y_i=g(X_i)+\epsilon_i\]</span> - We define <span class="math inline">\(g(x)\)</span> als the expected outcome for subjects with <span class="math inline">\(X_i=x\)</span></p>
<p><span class="math display">\[E[Y_i|X_i=x]=g(x)\]</span></p>
<p>Hence, <span class="math inline">\(\epsilon_i\)</span> is on average 0 for subjects with same <span class="math inline">\(X_i\)</span>: <span class="math display">\[E[\epsilon_i|X_i]=0\]</span></p>
</div>
<div id="lineair-regression-1" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Lineair regression</h2>
<ul>
<li>To obtain <em>accurate</em> and <em>interpretable</em> results one often choose <span class="math inline">\(g(x)\)</span> to be a linear function with unknown parameter.</li>
</ul>
<p><span class="math display">\[E(Y|X=x)=\beta_0 + \beta_1 x\]</span></p>
<p>unknown  <span class="math inline">\(\beta_0\)</span> and  <span class="math inline">\(\beta_1\)</span>.</p>
<ul>
<li><p>Lineair model imposes an <em>assumption</em> on the distribution of <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span>, which can be invalid.</p></li>
<li><p><em>Efficient data-analysis</em>: because it uses all observations to learn on the expected outcome for <span class="math inline">\(X=x\)</span>.</p></li>
</ul>
</div>
<div id="use" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Use</h2>
<ul>
<li><p><em>Prediction</em>: when <span class="math inline">\(Y\)</span> is unknown but <span class="math inline">\(X\)</span> is known we can predict <span class="math inline">\(Y\)</span> using <span class="math display">\[E(Y|X=x)=\beta_0 + \beta_1 x\]</span></p></li>
<li><p><em>Association</em>: biological relation between variable <span class="math inline">\(X\)</span> and response <span class="math inline">\(Y\)</span></p></li>
<li><p><em>Intercept:</em> <span class="math inline">\(E(Y|X=0)=\beta_0\)</span> </p></li>
<li><p><em>Slope</em>: <span class="math display">\[\begin{eqnarray*}
E(Y|X=x+\delta)-E(Y|X=x)&amp;=&amp;\beta_0 + \beta_1 (x+\delta) -\beta_0-\beta_1 x\\
&amp;=&amp; \beta_1\delta
\end{eqnarray*}\]</span></p></li>
</ul>
<p><span class="math inline">\(\beta_1=\)</span> difference in mean outcome for subjects that differ in one unit of the predictor <span class="math inline">\(X\)</span>.</p>
</div>
</div>
<div id="parameter-estimation" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Parameter estimation</h1>
<ul>
<li>Least squares</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>brcaSubset <span class="sc">%&gt;%</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ESR1,<span class="at">y=</span>S100A8)) <span class="sc">+</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">se=</span><span class="cn">FALSE</span>,<span class="at">col=</span><span class="st">&quot;grey&quot;</span>) <span class="sc">+</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">&quot;lm&quot;</span>,<span class="at">se=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li><p>Parameters <span class="math inline">\(\beta_0\)</span> en <span class="math inline">\(\beta_1\)</span> are unknown.</p></li>
<li><p>Estimate them using sample</p></li>
<li><p>Best fitting line</p>
<ul>
<li>Point on regression line for a given <span class="math inline">\(x_i\)</span>: <span class="math inline">\((x_i, \beta_0 + \beta_1 x_i)\)</span> as close as possible <span class="math inline">\((x_i, y_i)\)</span></li>
<li>Choose <span class="math inline">\(\beta_0\)</span> and <span class="math inline">\(\beta_1\)</span> so that the sum between predicted and observed points becomes as small as possible.</li>
</ul></li>
</ul>
<p><span class="math display">\[SSE=\sum_{i=1}^n (y_i-\beta_0-\beta_1 x_i)^2=\sum_{i=1}^n e_i^2\]</span></p>
<p>with residuals <span class="math inline">\(e_i\)</span> the vertical distances from the observations to the fitted regression line</p>
<div id="estimators-that-minimise-sse" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Estimators that minimise SSE</h2>
<p><span class="math display">\[\hat{\beta_1}= \frac{\sum\limits_{i=1}^n (y_i-\bar y)(x_i-\bar x)}{\sum\limits_{i=1}^n (x_i-\bar x_i)^2}=\frac{\mbox{cor}(x,y)s_y}{s_x} \]</span></p>
<p><span class="math display">\[\hat{\beta_0}=\bar y - \hat{\beta}_1 \bar x \]</span></p>
<p>Note, that the slope of the least squares fit is proportional to the correlation between the response and the predictor.</p>
<p>Fitted model allows to:</p>
<ul>
<li><p>predict the response for subjects with a given value <span class="math inline">\(x\)</span> for the predictor: <span class="math display">\[\text{E} [ Y | X = x]=\hat{\beta}_0+\hat{\beta}_1x\]</span></p></li>
<li><p>Assess how the mean response differs between two groups of subjects that differ <span class="math inline">\(\delta\)</span> units in the predictor:</p></li>
</ul>
<p><span class="math display">\[\text{E}\left[Y|X=x+\delta\right]-\text{E}\left[Y|X=x\right]= \hat{\beta}_1\delta\]</span></p>
<div id="breast-cancer-example" class="section level3" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> Breast cancer example</h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>lm1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(S100A8<span class="sc">~</span>ESR1,brcaSubset)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm1)</span></code></pre></div>
<pre><code>
Call:
lm(formula = S100A8 ~ ESR1, data = brcaSubset)

Residuals:
   Min     1Q Median     3Q    Max 
-95.43 -34.81  -6.79  34.23 145.21 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 208.47145   28.57207   7.296 7.56e-08 ***
ESR1         -0.05926    0.01212  -4.891 4.08e-05 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 59.91 on 27 degrees of freedom
Multiple R-squared:  0.4698,    Adjusted R-squared:  0.4502 
F-statistic: 23.93 on 1 and 27 DF,  p-value: 4.078e-05</code></pre>
<p><span class="math display">\[E(Y|X=x)=208.47-0.059 x\]</span></p>
<ul>
<li><p>Expected S100A8 expression is on average 59 units lower for patients with ESR1 expression level that is 1000 units higher</p></li>
<li><p>Expected S100A8 expression level for patients with an ESR1 expression level of 2000: <span class="math display">\[208.47-0.059\times 2000=89.94\]</span></p></li>
<li><p>Expected S100A8 expression level for patients with an ESR1 expression level of 4000: <span class="math display">\[208.47-0.059\times 4000=-28.58\]</span></p></li>
<li><p>Be careful when you extrapolate! (We can only assess the assumption of linearity within the range of the data).</p></li>
</ul>
</div>
</div>
</div>
<div id="statistical-inference" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Statistical inference</h1>
<p>To draw conclusions based on the regression model <span class="math display">\[E(Y|X)=\beta_0+\beta_1 X\]</span> we need to know</p>
<ul>
<li><p>How the least squares parameter estimators vary from sample to sample, and</p></li>
<li><p>how they deviate under the null hypothesis that there is no association between predictor and response</p></li>
<li><p>Requires a statistical model</p></li>
<li><p>Model the distribution of <span class="math inline">\(Y\)</span> given <span class="math inline">\(X\)</span> explicitly: f_{Y|X}(y)</p></li>
</ul>
<div id="modelling-distribution-of-y" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Modelling distribution of Y?</h2>
<ol style="list-style-type: decimal">
<li>Besides <em>Linearity</em> we need additional assumptions!</li>
<li><em>Independence</em>: Observations <span class="math inline">\((X_1,Y_1), ..., (X_n,Y_n)\)</span> are made for n independent subjects (is required to estimate the variance)</li>
<li><em>Homoscedasticity</em> or <em>equal variances</em>: observations vary with equal mean around the regression line
<ul>
<li>Residuals <span class="math inline">\(\epsilon_i\)</span> have equal variance for each <span class="math inline">\(X_i=x\)</span></li>
<li><span class="math inline">\(\text{var}(Y\vert X=x) = \sigma^2\)</span> for each <span class="math inline">\(X=x\)</span></li>
<li><span class="math inline">\(\sigma\)</span> is referred to as the <em>residual standard deviation</em></li>
</ul></li>
<li><em>Normality</em>: the residuals <span class="math inline">\(\epsilon_i\)</span> are normally distributed</li>
</ol>
<p><img src="https://raw.githubusercontent.com/GTPB/PSLS20/gh-pages/assets/figs/RegModel3.png" style="width:100.0%" /></p>
<ul>
<li><p>Given 2, 3 and 4 <span class="math display">\[\epsilon_i \text{ i.i.d.} N(0,\sigma^2).\]</span></p></li>
<li><p>Together with 1 this implies: <span class="math display">\[Y_i\vert X_i\sim N(\beta_0+\beta_1 X_i,\sigma^2),\]</span></p></li>
<li><p>We can show that given these assumption <span class="math display">\[\sigma^2_{\hat{\beta}_0}=\frac{\sum\limits_{i=1}^n X^2_i}{\sum\limits_{i=1}^n (X_i-\bar X)^2} \times\frac{\sigma^2}{n} \text{ en } \sigma^2_{\hat{\beta}_1}=\frac{\sigma^2}{\sum\limits_{i=1}^n (X_i-\bar X)^2}\]</span></p></li>
<li><p>and the parameter estimators are also normally distributed <span class="math display">\[\hat\beta_0 \sim N\left(\beta_0,\sigma^2_{\hat \beta_0}\right) \text{ en } \hat\beta_1 \sim N\left(\beta_1,\sigma^2_{\hat \beta_1}\right)\]</span></p></li>
</ul>
</div>
<div id="high-spread-of-x-improves-the-precision" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> High spread of <span class="math inline">\(X\)</span> improves the precision</h2>
<p><span class="math display">\[\sigma^2_{\hat{\beta}_1}=\frac{\sigma^2}{\sum\limits_{i=1}^n (X_i-\bar X)^2}\]</span></p>
<p><img src="https://raw.githubusercontent.com/GTPB/PSLS20/gh-pages/assets/figs/spread.png" style="width:100.0%" /></p>
<ul>
<li>Conditional variance (<span class="math inline">\(\sigma^2\)</span>) is unknown</li>
<li>Estimate using <em>mean squared error</em> (MSE) <span class="math display">\[\hat\sigma^2=MSE=\frac{\sum\limits_{i=1}^n \left(y_i-\hat\beta_0-\hat\beta_1\times x_i\right)^2}{n-2}=\frac{\sum\limits_{i=1}^n e^2_i}{n-2}.\]</span></li>
<li>This estimator is based on independence (assumption 2) and equality of the variance (assumption 3).</li>
<li>Devide by <span class="math inline">\(n-2\)</span></li>
</ul>
<p>Upon the estimation of <span class="math inline">\(\sigma^2\)</span> we obtain following standard errors:</p>
<p><span class="math display">\[\text{SE}_{\hat{\beta}_0}=\hat\sigma_{\hat{\beta}_0}=\sqrt{\frac{\sum\limits_{i=1}^n X^2_i}{\sum\limits_{i=1}^n (X_i-\bar X)^2} \times\frac{\text{MSE}}{n}} \text{ en } \text{SE}_{\hat{\beta}_1}=\hat\sigma_{\hat{\beta}_1}=\sqrt{\frac{\text{MSE}}{\sum\limits_{i=1}^n (X_i-\bar X)^2}}\]</span></p>
<ul>
<li><p>Again we can construct tests and confidence intervals using <span class="math display">\[T=\frac{\hat{\beta}_k-\beta_k}{SE(\hat{\beta}_k)} \text{ with } k=1,2.\]</span></p></li>
<li><p>If all assumptions are valid <span class="math inline">\(T\)</span> follows t-verdeling with n-2 degrees of freedom. </p></li>
<li><p>If no normality, but independence, linearity, equality of mean and large dataset <span class="math display">\[\rightarrow \text{Central Limit theorem}\]</span></p></li>
</ul>
<div id="breast-cancer-example-1" class="section level3" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Breast cancer example</h3>
<ul>
<li><p>Negative association between S100A8 and ESR1 gene expression.</p></li>
<li><p>Generalize effect in sample to population using the confidence interval on the mean: <span class="math display">\[[\hat\beta_1 - t_{n-2,\alpha/2} \text{SE}_{\hat\beta_1},\hat\beta_1 + t_{n-2,\alpha/2} \text{SE}_{\hat\beta_1}]\]</span>.</p></li>
</ul>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">confint</span>(lm1)</span></code></pre></div>
<pre><code>                   2.5 %       97.5 %
(Intercept) 149.84639096 267.09649989
ESR1         -0.08412397  -0.03440378</code></pre>
<ul>
<li>Negative association is significant on 5% significance level.</li>
</ul>
</div>
</div>
<div id="hypothesis-test" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Hypothesis test</h2>
<ul>
<li><p>Translate the research question to assess the association between the S100A8 and ESR1 gene expression to parameters in the model.</p></li>
<li><p>Under the null hypothesis of the absence of an association in the expression of both genes: <span class="math display">\[H_0: \beta_1=0\]</span></p></li>
<li><p>Under the alternative hypothesis, there is an association between the expression of both genes : <span class="math display">\[H_1: \beta_1\neq0\]</span> </p></li>
<li><p>Test statistic <span class="math display">\[T=\frac{\hat{\beta}_1-0}{SE(\hat{\beta}_k)}\]</span> </p></li>
<li><p>Under <span class="math inline">\(H_0\)</span> the statistics follows a t-distribution with n-2 degrees of freedom.</p></li>
</ul>
<div id="brca-dataset" class="section level3" number="4.3.1">
<h3><span class="header-section-number">4.3.1</span> BRCA dataset</h3>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm1)</span></code></pre></div>
<pre><code>
Call:
lm(formula = S100A8 ~ ESR1, data = brcaSubset)

Residuals:
   Min     1Q Median     3Q    Max 
-95.43 -34.81  -6.79  34.23 145.21 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 208.47145   28.57207   7.296 7.56e-08 ***
ESR1         -0.05926    0.01212  -4.891 4.08e-05 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 59.91 on 27 degrees of freedom
Multiple R-squared:  0.4698,    Adjusted R-squared:  0.4502 
F-statistic: 23.93 on 1 and 27 DF,  p-value: 4.078e-05</code></pre>
<ul>
<li>The association between the S100A8 and ESR1 expression is extremely significant (p&lt;&lt;0.001).</li>
<li>But, we first have to check all assumptions!</li>
<li>Otherwise the conclusions based on the statistical test and the CI can be incorrect.</li>
</ul>
</div>
</div>
</div>
<div id="assess-assumptions" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Assess assumptions</h1>
<ul>
<li>Independence: design</li>
<li>Linearity: inference is useless if the association is not linear</li>
<li>Homoscedasticity: inference/p-value is incorrect if data are heteroscedastic</li>
<li>Normality: inference/p-value is incorrect if data are not normally distributed in small samples</li>
</ul>
<div id="linearity" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Linearity</h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>brcaSubset <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ESR1,<span class="at">y=</span>S100A8)) <span class="sc">+</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">se=</span><span class="cn">FALSE</span>,<span class="at">col=</span><span class="st">&quot;grey&quot;</span>) <span class="sc">+</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">&quot;lm&quot;</span>,<span class="at">se=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="residual-analysis" class="section level3" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Residual analysis</h3>
<ul>
<li>Assumption of linearity is typically assessed using <em>residual plot</em>. (Especially if the lineair model has multiple covariates, later chapters)</li>
<li>predictor of predictions <span class="math inline">\(\hat\beta_0+\hat\beta_1 x\)</span> on <span class="math inline">\(X\)</span>-axis</li>
<li><em>residuals</em> on <span class="math inline">\(Y\)</span>-as <span class="math display">\[e_i=y_i-\hat{g}(x_i)=y_i-\hat\beta_0-\hat\beta_1\times x_i,\]</span></li>
</ul>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(lm1)</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-11-1.png" width="672" /><img src="06-linearRegression_files/figure-html/unnamed-chunk-11-2.png" width="672" /><img src="06-linearRegression_files/figure-html/unnamed-chunk-11-3.png" width="672" /><img src="06-linearRegression_files/figure-html/unnamed-chunk-11-4.png" width="672" /></p>
</div>
</div>
<div id="homoscedasticity-equal-variances" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Homoscedasticity (equal variances)</h2>
<ul>
<li><p>Residuals and squared residuals cary information on the residual variability</p></li>
<li><p>Association with predictors <span class="math inline">\(\rightarrow\)</span> indication of heteroscedasticity.</p></li>
<li><p>Scatterplot of <span class="math inline">\(e_i\)</span> vs <span class="math inline">\(x_i\)</span> or predictions <span class="math inline">\(\hat \beta_0+ \hat \beta_1 x_i\)</span>.</p></li>
<li><p>Scatterplot van standardized residual versus <span class="math inline">\(x_i\)</span> or predictions.</p></li>
</ul>
</div>
<div id="normality" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Normality</h2>
<ul>
<li><p>If the sample size is large the estimators are normally distributed even if the observations are not normally distributed: central limit theorem</p></li>
<li><p>How many observations? <span class="math inline">\(\rightarrow\)</span> depends on shape and magnitude of deviations</p></li>
<li><p>Assumption: Data are Normally distributed conditional on X: <span class="math display">\[Y_i\vert X_i\sim N(\beta_0+\beta_1X_i,\sigma^2)\]</span></p></li>
<li><p>QQ-plot of response Y is misleading and useless: distribution of <span class="math inline">\(Y_i\)</span> are different because they have a different conditional mean!</p></li>
<li><p>QQ-plot of the residuals <span class="math inline">\(e_i\)</span></p></li>
</ul>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(lm1,<span class="at">which=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
</div>
</div>
<div id="invalid-assumptions" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Invalid assumptions</h1>
<ul>
<li><p>Transformation of predictor does not change distribution of Y for given X:</p>
<ul>
<li>not useful to obtain homoscedasticity or Normal distribution</li>
<li>useful for linearity when normality and homoscedasticity are valid</li>
<li>Often inclusion of higher order terms: <span class="math inline">\(X^2\)</span>, <span class="math inline">\(X^3\)</span>, … <span class="math display">\[Y_i=\beta_0+\beta_1X_i+\beta_2X_i^2+ ... + \epsilon_i\]</span></li>
</ul></li>
<li><p>Transformation of response Y can be useful to obtain normality and homoscedasticity</p></li>
<li><p><span class="math inline">\(\sqrt(Y)\)</span>, <span class="math inline">\(\log(Y)\)</span>, 1/Y, …</p></li>
</ul>
<div id="breast-cancer-example-2" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Breast cancer example</h2>
<p>Problems with</p>
<ul>
<li>heteroscedasticity</li>
<li>possibly deviations from normality (skewed to the right)</li>
<li>negative concentration predictions are theoretically impossible</li>
<li>non-linearity</li>
</ul>
<p>This is often the case for concentration and intensity measurements</p>
<ul>
<li>These are often log-normal distributed (normal distribution upon log-transformatie)</li>
<li>We also observed a kind of exponential relation with the smoother</li>
<li>In gene expression literature often <span class="math inline">\(\log_2\)</span> transformation is adopted</li>
<li>gene-expression on log scale: differences on log scale are fold changes on original scale!</li>
</ul>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>brca <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ESR1,<span class="at">y=</span>S100A8)) <span class="sc">+</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>()</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>brca <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ESR1<span class="sc">%&gt;%</span>log2,<span class="at">y=</span>S100A8<span class="sc">%&gt;%</span>log2)) <span class="sc">+</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>()</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>lm2<span class="ot">&lt;-</span><span class="fu">lm</span>(S100A8<span class="sc">%&gt;%</span>log2 <span class="sc">~</span> ESR1 <span class="sc">%&gt;%</span> log2, brca)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(lm2)</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-16-1.png" width="672" /><img src="06-linearRegression_files/figure-html/unnamed-chunk-16-2.png" width="672" /><img src="06-linearRegression_files/figure-html/unnamed-chunk-16-3.png" width="672" /><img src="06-linearRegression_files/figure-html/unnamed-chunk-16-4.png" width="672" /></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm2)</span></code></pre></div>
<pre><code>
Call:
lm(formula = S100A8 %&gt;% log2 ~ ESR1 %&gt;% log2, data = brca)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.94279 -0.66537  0.08124  0.68468  1.92714 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)     23.401      1.603   14.60 3.57e-15 ***
ESR1 %&gt;% log2   -1.615      0.150  -10.76 8.07e-12 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 1.026 on 30 degrees of freedom
Multiple R-squared:  0.7942,    Adjusted R-squared:  0.7874 
F-statistic: 115.8 on 1 and 30 DF,  p-value: 8.07e-12</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">confint</span>(lm2)</span></code></pre></div>
<pre><code>                  2.5 %    97.5 %
(Intercept)   20.128645 26.674023
ESR1 %&gt;% log2 -1.921047 -1.308185</code></pre>
<div id="interpretation-1" class="section level3" number="6.1.1">
<h3><span class="header-section-number">6.1.1</span> Interpretation 1</h3>
<p>A patient with an ESR1 expression that is one unit on <span class="math inline">\(\log_2\)</span> scale higher than that of another patient on average has a <span class="math inline">\(\log_2\)</span> expression for S100A8 that is 1.61 units lower (95% CI [-1.92,-1.31]).</p>
<p><span class="math display">\[\log_2 \hat\mu_1=23.401  -1.615 \times \text{logESR}_1,\text{ } \log_2 \hat\mu_2=23.401  -1.615 \times \text{logESR}_2 \]</span> <span class="math display">\[\log_2 \hat\mu_2-\log_2 \hat\mu_1=  -1.615 (\log_2 \text{ESR}_2-\log_2 \text{ESR}_1) = -1.615 \times 1 = -1.615\]</span></p>
</div>
<div id="interpretation-2" class="section level3" number="6.1.2">
<h3><span class="header-section-number">6.1.2</span> Interpretation 2</h3>
<p>Model on log-scale: upon back-transformation we obtain geometric means</p>
<p><span class="math display">\[\begin{eqnarray*}
\sum\limits_{i=1}^n \frac{\log x_i}{n}&amp;=&amp;\frac{\log x_1 + \ldots + \log x_n}{n}\\\\
&amp;\stackrel{(1)}{=}&amp;\frac{\log(x_1 \times \ldots \times x_n)}{n}=\frac{\log\left(\prod\limits_{i=1}^n x_i\right)}{n}\\\\
&amp;\stackrel{(2)}{=}&amp;\log \left(\sqrt[\leftroot{-1}\uproot{2}\scriptstyle n]{\prod\limits_{i=1}^n x_i}\right)
\end{eqnarray*}\]</span></p>
<ul>
<li>Population mean <span class="math inline">\(\mu\)</span> is estimated as a geometric mean</li>
<li>Logarithmic transformation is monotone: we can backtransform confidence intervals on log-scale!</li>
</ul>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="dv">2</span><span class="sc">^</span>lm2<span class="sc">$</span>coef[<span class="dv">2</span>]</span></code></pre></div>
<pre><code>ESR1 %&gt;% log2 
    0.3265519 </code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="dv">2</span><span class="sc">^-</span>lm2<span class="sc">$</span>coef[<span class="dv">2</span>]</span></code></pre></div>
<pre><code>ESR1 %&gt;% log2 
       3.0623 </code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="dv">2</span><span class="sc">^-</span><span class="fu">confint</span>(lm2)[<span class="dv">2</span>,]</span></code></pre></div>
<pre><code>   2.5 %   97.5 % 
3.786977 2.476298 </code></pre>
<p>A patient with an ESR1 expression that is 2 times the expression of that of another patient will on average have an S100A8 expression that is 3.06 times lower (95% CI [2.48,3.79]).</p>
<p><span class="math display">\[\log_2 \hat\mu_1=23.401  -1.615 \times \text{logESR}_1,\text{ } \log_2 \hat\mu_2=23.401  -1.615 \times \text{logESR}_2 \]</span> <span class="math display">\[\log_2 \hat\mu_2-\log_2 \hat\mu_1=  -1.615 (\log_2 \text{ESR}_2-\log_2 \text{ESR}_1) \]</span> <span class="math display">\[\log_2 \left[\frac{\hat\mu_2}{\hat\mu_1}\right]=  -1.615 \log_2\left[\frac{ \text{ESR}_2}{\text{ESR}_1}\right] \]</span> <span class="math display">\[\frac{\hat\mu_2}{\hat\mu_1}=\left[\frac{ \text{ESR}_2}{\text{ESR}_1}\right]^{-1.615}=2^ {-1.615} =0.326\]</span> or <span class="math display">\[\frac{\hat\mu_1}{\hat\mu_2}=2^{1.615} =3.06\]</span></p>
</div>
<div id="interpretation-3" class="section level3" number="6.1.3">
<h3><span class="header-section-number">6.1.3</span> Interpretation 3</h3>
<p>A patient with an ESR1 expression that is 1% higher than that of another patient will on average have an expression-level for S100A8 gen that is approximately -1.61% lower (95% CI [-1.92,-1.31])%.</p>
<p><span class="math display">\[\log_2 \hat\mu_1=23.401  -1.615 \times \text{logESR}_1,\text{ } \log_2 \hat\mu_2=23.401  -1.615 \times \text{logESR}_2 \]</span> <span class="math display">\[\log_2 \hat\mu_2-\hat\log_2 \mu_1=  -1.615 (\log_2 \text{ESR}_2-\log_2 \text{ESR}_1) \]</span> <span class="math display">\[\log_2 \left[\frac{\hat\mu_2}{\hat\mu_1}\right]=  -1.615 \log_2\left[\frac{ \text{ESR}_2}{\text{ESR}_1}\right] \]</span> <span class="math display">\[\frac{\hat\mu_2}{\hat\mu_1}=\left[\frac{ \text{ESR}_2}{\text{ESR}_1}\right]^{-1.615}=1.01^ {-1.615} =0.984 \approx -1.6\%\]</span></p>
<p>This is valid for low to moderate values of <span class="math inline">\(\beta_1\)</span>: <span class="math display">\[-10&lt;\beta_1&lt;10 \rightarrow 1.01^{\beta_1} -1 \approx \frac{\beta_1}{100}.\]</span></p>
</div>
</div>
<div id="inference-on-the-mean-outcome" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Inference on the mean outcome</h2>
<ul>
<li>A regression model can also be used for prediction</li>
<li>Inference on average outcome for a given value of <span class="math inline">\(X=x\)</span>, i.e. <span class="math display">\[\hat{g}(x)= \hat{\beta}_0 + \hat{\beta}_1 x\]</span></li>
<li><span class="math inline">\(\hat{g}(x)\)</span> is an estimator of the conditional mean <span class="math inline">\(E[Y\vert X=x]\)</span></li>
<li>Parameter estimators are Normally distributed and unbiased <span class="math inline">\(\rightarrow\)</span> estimator <span class="math inline">\(\hat{g}(x)\)</span> is also Normally distributed and unbiased.</li>
</ul>
<p><span class="math display">\[\text{SE}_{\hat{g}(x)}=\sqrt{MSE\left\{\frac{1}{n}+\frac{(x-\bar X)^2}{\sum\limits_{i=1}^n (X_i-\bar X)^2}\right\}}.\]</span></p>
<p><span class="math display">\[T=\frac{\hat{g}(x)-g(x)}{SE_{\hat{g}(x)}}\sim t_{n-2}\]</span></p>
<ul>
<li>Mean response and confidence intervals for the mean response in R via de <code>predict(.)</code> functie.</li>
<li><code>newdata</code> argument: predictor values (x-values) at which we want to calculate the mean response</li>
<li><code>interval="confidence"</code> argument to obtain CI.</li>
<li>Without newdata argument we perform predictions for all predictor values in the dataset used to fit the model.</li>
</ul>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>grid <span class="ot">&lt;-</span> <span class="dv">140</span><span class="sc">:</span><span class="dv">4000</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">predict</span>(lm2,<span class="at">newdata=</span><span class="fu">data.frame</span>(<span class="at">ESR1=</span>grid), <span class="at">interval=</span><span class="st">&quot;confidence&quot;</span>)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(g)</span></code></pre></div>
<pre><code>       fit      lwr      upr
1 11.89028 10.76082 13.01974
2 11.87370 10.74721 13.00019
3 11.85724 10.73370 12.98078
4 11.84089 10.72028 12.96151
5 11.82466 10.70696 12.94237
6 11.80854 10.69372 12.92336</code></pre>
<p>Note, that we do not have to transform the new data that we specified for the ESR1 expression because we fitted the model with a call to the <code>lm</code> function and specified the transformation within the lm formula using the pipe command!</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>brca <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ESR1<span class="sc">%&gt;%</span>log2,<span class="at">y=</span>S100A8<span class="sc">%&gt;%</span>log2)) <span class="sc">+</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">&quot;lm&quot;</span>)</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
</div>
<div id="back-transformation" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Back-transformation</h2>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>newdata<span class="ot">&lt;-</span><span class="fu">data.frame</span>(<span class="fu">cbind</span>(grid,<span class="dv">2</span><span class="sc">^</span>g))</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>brca <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ESR1,<span class="at">y=</span>S100A8)) <span class="sc">+</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>grid,<span class="at">y=</span>fit),newdata) <span class="sc">+</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>grid,<span class="at">y=</span>lwr),newdata,<span class="at">color=</span><span class="st">&quot;grey&quot;</span>) <span class="sc">+</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>grid,<span class="at">y=</span>upr),newdata,<span class="at">color=</span><span class="st">&quot;grey&quot;</span>)</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
</div>
</div>
<div id="prediction-intervals" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Prediction-intervals</h1>
<ul>
<li><p>We can also make a prediction for the location of a new observation that would be collected in a new experiment for a patient with a particular value for their ESR1 expression</p></li>
<li><p>It is important to notice that this experiment still has to be conducted. So we want to predict the non-observed individual expression value for a novel patient.</p></li>
<li><p>For a novel independent observation <span class="math inline">\(Y^*\)</span> <span class="math display">\[
Y^* = g(x) + \epsilon^*
\]</span> with <span class="math inline">\(\epsilon^*\sim N(0,\sigma^2)\)</span> and <span class="math inline">\(\epsilon^*\)</span> independent of the observations in the sample <span class="math inline">\(Y_1,\ldots, Y_n\)</span>.</p></li>
<li><p>We predict a new log-S100A8 for a patient with a known log2-ESR1 expression level x <span class="math display">\[
\hat{y}(x)=\hat{\beta}_0+\hat{\beta}_1 \times x
\]</span></p></li>
<li><p>The estimated mean outcome and prediction for a new observation are equal.</p></li>
<li><p>But, their sample distributions are different!</p>
<ul>
<li>Uncertainty on the estimated mean outcome <span class="math inline">\(\leftarrow\)</span> uncertainty on estimated model parameters <span class="math inline">\(\hat\beta_0\)</span> en <span class="math inline">\(\hat\beta_1\)</span>.</li>
<li>Uncertainty on new observation $ <span class="math inline">\(\leftarrow\)</span> <em>uncertainty on estimated mean</em> and <em>additional uncertainty</em> because the new observation will deviate around the mean!</li>
</ul></li>
</ul>
<p><span class="math display">\[\text{SE}_{\hat{Y}(x)}=\sqrt{\hat\sigma^2+\hat\sigma^2_{\hat{g}(x)}}=\sqrt{MSE\left\{1+\frac{1}{n}+\frac{(x-\bar X)^2}{\sum\limits_{i=1}^n (X_i-\bar X)^2}\right\}}.\]</span></p>
<p><span class="math display">\[\frac{\hat{Y}(x)-Y}{\text{SE}_{\hat{Y}(x)}}\sim t_{n-2}\]</span></p>
<ul>
<li>Note, that a <strong>prediction-interval</strong> (PI) is an improved version of a reference-interval when the model parameters are unknown: Uncertainty on model parameters + t-distribution.</li>
</ul>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">predict</span>(lm2,<span class="at">newdata=</span><span class="fu">data.frame</span>(<span class="at">ESR1=</span>grid), <span class="at">interval=</span><span class="st">&quot;prediction&quot;</span>)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(p)</span></code></pre></div>
<pre><code>       fit      lwr      upr
1 11.89028 9.510524 14.27004
2 11.87370 9.495354 14.25205
3 11.85724 9.480288 14.23419
4 11.84089 9.465324 14.21646
5 11.82466 9.450461 14.19886
6 11.80854 9.435698 14.18138</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>preddata<span class="ot">&lt;-</span><span class="fu">data.frame</span>(<span class="fu">cbind</span>(<span class="at">grid=</span>grid<span class="sc">%&gt;%</span>log2,p))</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>brca <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ESR1<span class="sc">%&gt;%</span>log2,<span class="at">y=</span>S100A8<span class="sc">%&gt;%</span>log2)) <span class="sc">+</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">&quot;lm&quot;</span>) <span class="sc">+</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>     <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>grid,<span class="at">y=</span>lwr),preddata,<span class="at">color=</span><span class="st">&quot;blue&quot;</span>) <span class="sc">+</span></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>grid,<span class="at">y=</span>upr),preddata,<span class="at">color=</span><span class="st">&quot;blue&quot;</span>)</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>preddata<span class="ot">&lt;-</span><span class="fu">data.frame</span>(<span class="fu">cbind</span>(grid,<span class="dv">2</span><span class="sc">^</span>p))</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>brca <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>ESR1,<span class="at">y=</span>S100A8)) <span class="sc">+</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>grid,<span class="at">y=</span>fit),newdata) <span class="sc">+</span></span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>grid,<span class="at">y=</span>lwr),newdata,<span class="at">color=</span><span class="st">&quot;grey&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>grid,<span class="at">y=</span>upr),newdata,<span class="at">color=</span><span class="st">&quot;grey&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>grid,<span class="at">y=</span>lwr),preddata,<span class="at">color=</span><span class="st">&quot;blue&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>grid,<span class="at">y=</span>upr),preddata,<span class="at">color=</span><span class="st">&quot;blue&quot;</span>)</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
<div id="nhanes-example" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> NHANES example</h2>
<ul>
<li><p>Replace reference interval for cholesterol level from chapter 2 by prediction-interval.</p></li>
<li><p>Reference interval</p></li>
</ul>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(NHANES)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>fem <span class="ot">&lt;-</span> NHANES <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Gender<span class="sc">==</span><span class="st">&quot;female&quot;</span><span class="sc">&amp;!</span><span class="fu">is.na</span>(DirectChol))</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="fu">exp</span>(fem<span class="sc">$</span>DirectChol<span class="sc">%&gt;%</span>log<span class="sc">%&gt;%</span>mean <span class="sc">+</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">*</span> <span class="fu">qnorm</span>(<span class="fl">0.975</span>) <span class="sc">*</span> (fem<span class="sc">$</span>DirectChol<span class="sc">%&gt;%</span>log<span class="sc">%&gt;%</span>sd))</span></code></pre></div>
<pre><code>[1] 0.8361311 2.4397130</code></pre>
<ul>
<li>prediction interval</li>
</ul>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>lmChol <span class="ot">&lt;-</span> <span class="fu">lm</span>(DirectChol <span class="sc">%&gt;%</span> log2<span class="sc">~</span><span class="dv">1</span>,<span class="at">data=</span>fem)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>predInt <span class="ot">&lt;-</span> <span class="fu">predict</span>(lmChol,<span class="at">interval=</span><span class="st">&quot;prediction&quot;</span>,<span class="at">newdata=</span><span class="fu">data.frame</span>(<span class="at">noPred=</span><span class="dv">1</span>))</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="dv">2</span><span class="sc">^</span>predInt,<span class="dv">2</span>)</span></code></pre></div>
<pre><code>   fit  lwr  upr
1 1.43 0.84 2.44</code></pre>
<p>Note, that the prediction interval is almost similar to the reference interval for the large sample. Indeed we could estimate the parameters very precise.</p>
<p>We will do the same thing for the small sample size of 10 patients.</p>
<ul>
<li>Reference interval</li>
</ul>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>fem10<span class="ot">&lt;-</span> NHANES <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Gender<span class="sc">==</span><span class="st">&quot;female&quot;</span><span class="sc">&amp;!</span><span class="fu">is.na</span>(DirectChol)) <span class="sc">%&gt;%</span> <span class="fu">sample_n</span>(<span class="at">size=</span><span class="dv">10</span>)</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a><span class="dv">2</span><span class="sc">^</span>(fem10<span class="sc">$</span>DirectChol<span class="sc">%&gt;%</span>log2<span class="sc">%&gt;%</span>mean <span class="sc">+</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>)<span class="sc">*</span> <span class="fu">qnorm</span>(<span class="fl">0.975</span>) <span class="sc">*</span> (fem10<span class="sc">$</span>DirectChol<span class="sc">%&gt;%</span>log2<span class="sc">%&gt;%</span>sd))</span></code></pre></div>
<pre><code>[1] 0.8976012 2.2571645</code></pre>
<ul>
<li>Prediction interval</li>
</ul>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>lmChol10 <span class="ot">&lt;-</span> <span class="fu">lm</span>(DirectChol <span class="sc">%&gt;%</span> log2<span class="sc">~</span><span class="dv">1</span>,<span class="at">data=</span>fem10)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>predInt10 <span class="ot">&lt;-</span> <span class="fu">predict</span>(lmChol10,<span class="at">interval=</span><span class="st">&quot;prediction&quot;</span>,<span class="at">newdata=</span><span class="fu">data.frame</span>(<span class="at">noPred=</span><span class="dv">1</span>))</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="dv">2</span><span class="sc">^</span>predInt10,<span class="dv">2</span>)</span></code></pre></div>
<pre><code>   fit  lwr  upr
1 1.42 0.81 2.49</code></pre>
<ul>
<li><p>Note, that the PI now captures uncertainty in parameter estimators (mean and standard error). And that the interval becomes much wider! This is particularly important here for the upper limit because we back-transformed the data!</p></li>
<li><p>The interval is almost as wide as the one based on the large sample.</p></li>
<li><p>In small samples it is very important to account for this additional uncertainty.</p></li>
</ul>
</div>
</div>
<div id="sum-of-squares-and-anova-table" class="section level1" number="8">
<h1><span class="header-section-number">8</span> Sum of squares and Anova-table</h1>
<p>##Total sum of squares <span class="math display">\[\text{SSTot} = \sum_{i=1}^n (Y_i-\bar{Y})^2.\]</span></p>
<ul>
<li><p>SStot can be used to estimate the variance of the <strong>marginal distribution</strong> of the response.</p></li>
<li><p>In this chapter we focused on the <strong>conditional distribution</strong> <span class="math inline">\(f(Y\vert X=x)\)</span>.</p></li>
<li><p>We known that MSE is a good estimate of the variance of the conditional distribution of <span class="math inline">\(Y\vert X=x\)</span>.</p></li>
</ul>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-29-1.png" width="100%" style="display: block; margin: auto;" /></p>
<div id="sum-of-squares-of-the-regression-ssr" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Sum of squares of the regression SSR</h2>
<p><span class="math display">\[\text{SSR} = \sum_{i=1}^n (\hat{Y}_i - \bar{Y})^2 = \sum_{i=1}^n (\hat{g}(x_i) - \bar{Y})^2.\]</span></p>
<ul>
<li><p>Is a measure for the deviation of the predictions on the regression line and the marginal mean of the response.</p></li>
<li><p>Another interpretation: difference between two models</p>
<ul>
<li>Estimated model <span class="math inline">\(\hat{g}(x)=\hat\beta_0+\hat\beta_1x\)</span></li>
<li>Estimated model without predictor (only intercept): <span class="math inline">\(g(x)=\beta_0\)</span> <span class="math inline">\(\rightarrow\)</span> <span class="math inline">\(\beta_0\)</span> will be equal to <span class="math inline">\(\bar{Y}\)</span>.</li>
</ul></li>
<li><p>SSR measures the size of the effect of the predictor</p></li>
</ul>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-30-1.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="sum-of-squares-of-the-error" class="section level2" number="8.2">
<h2><span class="header-section-number">8.2</span> Sum of Squares of the Error</h2>
<p><span class="math display">\[ \text{SSE} = \sum_{i=1}^n (Y_i-\hat{Y}_i )^2 = \sum_{i=1}^n \left\{Y_i-\hat{g}\left(x_i\right)\right\}^2.\]</span></p>
<ul>
<li><p>The smaller SSE the better the fit.</p></li>
<li><p>Least squares method!</p></li>
</ul>
<hr />
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-31-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>We can show that SST can be decomposed in <span class="math display">\[\begin{eqnarray*}
  \text{SSTot}
    &amp;=&amp;  \sum_{i=1}^n (Y_i-\bar{Y})^2 \\
    &amp;=&amp;  \sum_{i=1}^n (Y_i-\hat{Y}_i+\hat{Y}_i-\bar{Y})^2 \\
    &amp;=&amp;  \sum_{i=1}^n (Y_i-\hat{Y}_i)^2+\sum_{i=1}^n(\hat{Y}_i-\bar{Y})^2 \\
    &amp;=&amp;  \text{SSE }+\text{SSR}
  \end{eqnarray*}\]</span></p>
<ul>
<li>Total variability in the data (SSTot) is partially explained by the predictor (SSR).</li>
<li>Variability that we cannot explain with the regression model is the residual variability (SSE).</li>
</ul>
</div>
<div id="determination-coefficient" class="section level2" number="8.3">
<h2><span class="header-section-number">8.3</span> Determination coefficient</h2>
<p><span class="math display">\[ R^2 = 1-\frac{\text{SSE}}{\text{SSTot}}=\frac{\text{SSR}}{\text{SSTot}}.\]</span></p>
<ul>
<li><p><em>Fraction of total variability of the sample outcomes explained by the model</em>.</p></li>
<li><p>Large <span class="math inline">\(R^2\)</span> indicates that the model has the potential to make good predictions (small SSE).</p></li>
<li><p>Not very indicative for p-value of the test <span class="math inline">\(H_0:\beta_1=0\)</span> vs <span class="math inline">\(H_1:\beta_1\neq0\)</span>.</p>
<ul>
<li>p-value is largely determined by SSE and sample size <span class="math inline">\(n\)</span>, but not by SSTot.</li>
<li><span class="math inline">\(R^2\)</span> is determined by SSE and SSTot but not by sample size <span class="math inline">\(n\)</span>.</li>
</ul></li>
<li><p>Model with low <span class="math inline">\(R^2\)</span> is still useful to study associations as long as the association is modelled correctly!</p></li>
</ul>
<div id="breast-cancer-example-3" class="section level3" number="8.3.1">
<h3><span class="header-section-number">8.3.1</span> Breast cancer example</h3>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm2)</span></code></pre></div>
<pre><code>
Call:
lm(formula = S100A8 %&gt;% log2 ~ ESR1 %&gt;% log2, data = brca)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.94279 -0.66537  0.08124  0.68468  1.92714 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)     23.401      1.603   14.60 3.57e-15 ***
ESR1 %&gt;% log2   -1.615      0.150  -10.76 8.07e-12 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 1.026 on 30 degrees of freedom
Multiple R-squared:  0.7942,    Adjusted R-squared:  0.7874 
F-statistic: 115.8 on 1 and 30 DF,  p-value: 8.07e-12</code></pre>
</div>
</div>
<div id="f-test-in-simple-linear-model" class="section level2" number="8.4">
<h2><span class="header-section-number">8.4</span> F-Test in simple linear model</h2>
<ul>
<li>Sum of squares are the bases for <span class="math inline">\(F\)</span>-tests <span class="math display">\[  F  = \frac{\text{MSR}}{\text{MSE}}\]</span></li>
</ul>
<p>with <span class="math inline">\(\text{MSR} = \frac{\text{SSR}}{1} \text{ and } \text{MSE} = \frac{\text{SSE}}{n-2}.\)</span></p>
<ul>
<li><p>MSR mean sum of squares of the regression,</p></li>
<li><p>denominators 1 en <span class="math inline">\(n-2\)</span> are the degrees of freedom of SSR and SSE.</p></li>
<li><p>Under <span class="math inline">\(H_0: \beta_1=0\)</span> <span class="math display">\[H_0:F = \frac{\text{MSR}}{\text{MSE}} \sim F_{1,n-2},\]</span></p></li>
<li><p>F-test is always two-sided! <span class="math inline">\(H_1:\beta_1\neq 0\)</span> <span class="math display">\[  p = P_0\left[F\geq f\right]=1-F_F(f;1,n-2)\]</span></p></li>
</ul>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm2)</span></code></pre></div>
<pre><code>
Call:
lm(formula = S100A8 %&gt;% log2 ~ ESR1 %&gt;% log2, data = brca)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.94279 -0.66537  0.08124  0.68468  1.92714 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)     23.401      1.603   14.60 3.57e-15 ***
ESR1 %&gt;% log2   -1.615      0.150  -10.76 8.07e-12 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 1.026 on 30 degrees of freedom
Multiple R-squared:  0.7942,    Adjusted R-squared:  0.7874 
F-statistic: 115.8 on 1 and 30 DF,  p-value: 8.07e-12</code></pre>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-34-1.png" width="672" /></p>
</div>
<div id="anova-table" class="section level2" number="8.5">
<h2><span class="header-section-number">8.5</span> Anova Table</h2>
<table>
<thead>
<tr class="header">
<th></th>
<th>Df</th>
<th>Sum Sq</th>
<th>Mean Sq</th>
<th>F value</th>
<th>Pr(&gt;F)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Regression</td>
<td>degrees of freedom SSR</td>
<td>SSR</td>
<td>MSR</td>
<td>f-statistic</td>
<td>p-value</td>
</tr>
<tr class="even">
<td>Error</td>
<td>degrees of freedom SSE</td>
<td>SSE</td>
<td>MSE</td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(lm2)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Df"],"name":[1],"type":["int"],"align":["right"]},{"label":["Sum Sq"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Mean Sq"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["F value"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Pr(>F)"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"1","2":"121.81364","3":"121.813644","4":"115.7978","5":"8.070269e-12","_rn_":"ESR1 %>% log2"},{"1":"30","2":"31.55854","3":"1.051951","4":"NA","5":"NA","_rn_":"Residuals"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div id="dummy-variables" class="section level1" number="9">
<h1><span class="header-section-number">9</span> Dummy variables</h1>
<ul>
<li><p>Linear regression model can also be used to compare two group means.</p></li>
<li><p>brca: difference in average age between patients with unaffected and affected lymph nodes.</p></li>
<li><p>Define dummy variabele <span class="math display">\[x_i = \left\{ \begin{array}{ll}
1 &amp; \text{affected lymph nodes} \\
0 &amp; \text{unaffected lymph nodes} \end{array}\right.\]</span></p></li>
<li><p>group with <span class="math inline">\(x_i=0\)</span> is referred to as the <strong>reference group</strong>.</p></li>
<li><p>Regression model remains unaltered, <span class="math display">\[Y_i = \beta_0 + \beta_1 x_i +\epsilon_i\]</span> with <span class="math inline">\(\epsilon_i \text{ iid } N(0,\sigma^2)\)</span></p></li>
</ul>
<p>Because <span class="math inline">\(x_i\)</span> only can take two values, we can study the regression model for each value of <span class="math inline">\(x_i\)</span> separately: <span class="math display">\[ \begin{array}{lcll}
   Y_i &amp;=&amp; \beta_0 +\epsilon_i &amp;\text{unaffected lymph nodes} (x_i=0) \\
   Y_i &amp;=&amp; \beta_0 + \beta_1 +\epsilon_i &amp;\text{ affected lymph nodes} (x_i=1) .
 \end{array}\]</span> So <span class="math display">\[\begin{eqnarray*}
   E\left[Y_i\mid x_i=0\right] &amp;=&amp; \beta_0 \\
   E\left[Y_i\mid x_i=1\right] &amp;=&amp; \beta_0 + \beta_1,
\end{eqnarray*}\]</span></p>
<p>Hence, the interpretation of <span class="math inline">\(\beta_1\)</span>: <span class="math display">\[   \beta_1 = E\left[Y_i\mid x_i=1\right]-E\left[Y_i\mid x_i=0\right]\]</span></p>
<p><span class="math inline">\(\beta_1\)</span> is the average age difference between patients with affected and patients with unaffected lymph nodes (reference group).</p>
<p>With notation <span class="math inline">\(\mu_0= E\left[Y_i\mid x_i=0\right]\)</span> and <span class="math inline">\(\mu_1= E\left[Y_i\mid x_i=1\right]\)</span> this becomes <span class="math display">\[\beta_1 = \mu_1-\mu_0.\]</span></p>
<p>We can show that <span class="math display">\[\begin{array}{ccll}
 \hat\beta_0
   &amp;=&amp; \bar{Y}_1&amp;\text{ (sample mean of reference group)} \\
 \hat\beta_1
   &amp;=&amp; \bar{Y}_2-\bar{Y}_1&amp;\text{(estimator of effect size)} \\
 \text{MSE}
   &amp;=&amp; S_p^2 .
\end{array}\]</span></p>
<p>Tests <span class="math inline">\(H_0:\beta_1=0\)</span> vs. <span class="math inline">\(H_1:\beta_1\neq0\)</span> can be used to assess the null hypothesis of the two-sample <span class="math inline">\(t\)</span>-test, <span class="math inline">\(H_0:\mu_1=\mu_2\)</span> vs <span class="math inline">\(H_1:\mu_1\neq\mu_2\)</span>.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>brca<span class="sc">$</span>node <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(brca<span class="sc">$</span>node)</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(age<span class="sc">~</span>node,brca,<span class="at">var.equal=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>
    Two Sample t-test

data:  age by node
t = -2.7988, df = 30, p-value = 0.008879
alternative hypothesis: true difference in means between group 0 and group 1 is not equal to 0
95 percent confidence interval:
 -15.791307  -2.467802
sample estimates:
mean in group 0 mean in group 1 
       59.94737        69.07692 </code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>lm3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(age<span class="sc">~</span>node,brca)</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lm3)</span></code></pre></div>
<pre><code>
Call:
lm(formula = age ~ node, data = brca)

Residuals:
     Min       1Q   Median       3Q      Max 
-19.9474  -5.3269   0.0526   5.3026  18.0526 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   59.947      2.079  28.834  &lt; 2e-16 ***
node1          9.130      3.262   2.799  0.00888 ** 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 9.063 on 30 degrees of freedom
Multiple R-squared:  0.207, Adjusted R-squared:  0.1806 
F-statistic: 7.833 on 1 and 30 DF,  p-value: 0.008879</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(lm3)</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-38-1.png" width="672" /><img src="06-linearRegression_files/figure-html/unnamed-chunk-38-2.png" width="672" /><img src="06-linearRegression_files/figure-html/unnamed-chunk-38-3.png" width="672" /><img src="06-linearRegression_files/figure-html/unnamed-chunk-38-4.png" width="672" /></p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>brca <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>node<span class="sc">%&gt;%</span>as.factor,<span class="at">y=</span>age)) <span class="sc">+</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-39-1.png" width="672" /></p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">3</span>))</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">354</span>)</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>) <span class="fu">plot</span>(<span class="fu">rnorm</span>(<span class="dv">32</span>)<span class="sc">~</span>node,brca,<span class="at">ylab=</span><span class="st">&quot;iid N(0,1)&quot;</span>)</span></code></pre></div>
<p><img src="06-linearRegression_files/figure-html/unnamed-chunk-40-1.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="observational-study" class="section level1" number="10">
<h1><span class="header-section-number">10</span> Observational study</h1>
<ul>
<li><p>We cannot conclude that age causes a higher risk for affected lymph nodes.</p></li>
<li><p>Possibly <strong>confounding</strong>: no randomisation <span class="math inline">\(\rightarrow\)</span> groups of patients with affected and unaffected lymph nodes. They can also differ in other characteristics.</p></li>
<li><p>We can only conclude that there is an association between lymph node status and age.</p></li>
<li><p>However, the association does not have to be causal!</p></li>
<li><p>Note, that this is also the case for the linear model for <span class="math inline">\(\log_2\)</span>-S100A8-expression.</p>
<ul>
<li>Because we were not able to fix the ESR1-expression experimentally we cannot conclude that a higher ESR1-expression causes a decrease in the S100A8-expression.</li>
<li>We can only conclude that there is a negative association.</li>
<li>To assess the impact of a gene on other gene typically knockout mutants are used in the lab.</li>
</ul></li>
</ul>
</div>

<div id="rmd-source-code">---
title: "6. Simple linear regression"
author: "Lieven Clement"
date: "statOmics, Ghent University (https://statomics.github.io)"
---

<a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a>

```{r setup, include=FALSE}
knitr::opts_chunk$set(include = TRUE, comment = NA, echo = TRUE,
                      message = FALSE, warning = FALSE, cache = TRUE)
library(Rmisc)
library(tidyverse)
```


# Breast cancer dataset


- Subset of study https://doi.org/10.1093/jnci/djj052

- 32 breast cancer patients with estrogen recepter positieve tumor that had tamoxifen chemotherapy. Variabels:

    - grade: histological grade of tumor (grade 1 vs 3),
    - node: lymph node status  (0: not affected, 1: lymph nodes affected and removed),
    - size: tumor size in cm,
    - ESR1 and S100A8 gene expression in tumor biopsy (microarray technology)


```{r}
brca <- read_csv("https://raw.githubusercontent.com/GTPB/PSLS20/master/data/breastcancer.csv")
brca
```

- For didactical reasons we first remove 3 outliers in the S100A8 expression data.
- Later in the lecture we will show how to properly deal with all data.


```{r out.width='70%', fig.align='center',warnings=FALSE}
brca %>% ggplot(aes(x="",y=S100A8)) +
geom_boxplot()
```

---

```{r}
library(GGally)
brcaSubset<-brca %>% filter(S100A8<2000)
brcaSubset[,-(1:4)] %>% ggpairs()
```

## Association between ESR1 and S100A8 expressie

- ESR1 in $\pm$ 75% of breast cancer tumors.

    - Expression of ER gene positive for treatment: tumor responds to hormone therapy
    - Tamoxifen interacts with ER and modulates gene expression.

- Proteins of S100 family often dysregulated in cancer

    - S100A8 expressie represses immune systeem in tumor en creates an environment of inflamation that promotes tumor growth.

- Assess association between ESR1 and S100A8 expression.

1. pipe dataset to ggplot
2. select data `ggplot(aes(x=ESR1,y=S100A8))`
3. add points `geom_point()`
4. add smooth line `geom_smooth()`

```{r fig.align='center'}
brcaSubset %>%
  ggplot(aes(x=ESR1,y=S100A8)) +
  geom_point() +
  geom_smooth()
```

# Lineair Regression

- Statistical method to assess association between two variables $(X_i, Y_i)$, measured on each subject $i = 1, ..., n$.

- Gene expression example

    - Response Y : S100A8 expression
    - Predictor X: ESR1 expression

```{r fig.align='center'}
brcaSubset %>%
  ggplot(aes(x=ESR1,y=S100A8)) +
  geom_point() +
  geom_smooth(se=FALSE,col="grey") +
  geom_smooth(method="lm",se=FALSE)
```

## Model

- For fixed $X$, $Y$ does not necessarly has the same value

$$\text{observation = signal + noise}$$

$$Y_i=g(X_i)+\epsilon_i$$
- We define $g(x)$ als the expected outcome for subjects with $X_i=x$

$$E[Y_i|X_i=x]=g(x)$$

Hence, $\epsilon_i$ is on average 0 for subjects with same  $X_i$:
$$E[\epsilon_i|X_i]=0$$

## Lineair regression

- To obtain *accurate* and *interpretable* results one often choose $g(x)$ to be a linear function with unknown parameter.

$$E(Y|X=x)=\beta_0 + \beta_1 x$$

unknown \alert{intercept} $\beta_0$ and
\alert{slope} $\beta_1$.

- Lineair model imposes an *assumption* on the distribution of $X$ and $Y$, which can be invalid.

- *Efficient data-analysis*: because it uses all observations to learn on the expected outcome for $X=x$.


## Use

- *Prediction*: when $Y$ is unknown but $X$ is known we can predict $Y$ using
\[E(Y|X=x)=\beta_0 + \beta_1 x\]

- *Association*: biological relation between variable $X$ and response $Y$
- *Intercept:* $E(Y|X=0)=\beta_0$
\vspace{10pt}
- *Slope*:
\begin{eqnarray*}
E(Y|X=x+\delta)-E(Y|X=x)&=&\beta_0 + \beta_1 (x+\delta) -\beta_0-\beta_1 x\\
&=& \beta_1\delta
\end{eqnarray*}

$\beta_1=$ difference in mean outcome for subjects that differ in one unit of the predictor  $X$.

# Parameter estimation

- Least squares

```{r fig.align='center'}
brcaSubset %>%
  ggplot(aes(x=ESR1,y=S100A8)) +
  geom_point() +
  geom_smooth(se=FALSE,col="grey") +
  geom_smooth(method="lm",se=FALSE)
```

- Parameters $\beta_0$ en $\beta_1$ are unknown.

- Estimate them using sample

- Best fitting line

    - Point on regression line for a given $x_i$: $(x_i, \beta_0 + \beta_1 x_i)$ as close as possible $(x_i, y_i)$
    - Choose $\beta_0$ and $\beta_1$ so that the sum between predicted and observed points becomes as small as possible.

$$SSE=\sum_{i=1}^n (y_i-\beta_0-\beta_1 x_i)^2=\sum_{i=1}^n e_i^2$$

with residuals $e_i$ the vertical distances from the observations to the fitted regression line

## Estimators that minimise SSE

$$\hat{\beta_1}= \frac{\sum\limits_{i=1}^n (y_i-\bar y)(x_i-\bar x)}{\sum\limits_{i=1}^n (x_i-\bar x_i)^2}=\frac{\mbox{cor}(x,y)s_y}{s_x} $$

$$\hat{\beta_0}=\bar y - \hat{\beta}_1 \bar x $$

Note, that the slope of the least squares fit is proportional to the correlation between the response and the predictor.

Fitted model allows to:

  - predict the response for subjects with a given value $x$ for the predictor:
$$\text{E} [ Y | X = x]=\hat{\beta}_0+\hat{\beta}_1x$$

  - Assess how the mean response differs between two groups of subjects that differ $\delta$ units in the predictor:

$$\text{E}\left[Y|X=x+\delta\right]-\text{E}\left[Y|X=x\right]= \hat{\beta}_1\delta$$

### Breast cancer example

```{r}
lm1 <- lm(S100A8~ESR1,brcaSubset)
summary(lm1)
```

\[E(Y|X=x)=`r round(lm1$coef[1],2)`-`r abs(round(lm1$coef[2],3))` x\]

- Expected S100A8 expression is on average  `r abs(round(lm1$coef[2],3)*1000)` units lower for patients with ESR1 expression level that is 1000 units higher

- Expected S100A8 expression level for patients with an ESR1 expression level of 2000:
\[`r round(lm1$coef[1],2)`-`r abs(round(lm1$coef[2],3))`\times 2000=`r round(lm1$coef[1]+lm1$coef[2]*2000,2)`\]

- Expected S100A8 expression level for patients with an ESR1 expression level of 4000:
\[`r round(lm1$coef[1],2)`-`r abs(round(lm1$coef[2],3))`\times 4000=`r round(lm1$coef[1]+lm1$coef[2]*4000,2)`\]
- Be careful when you extrapolate! (We can only assess the assumption of linearity within the range of the data).

# Statistical inference

To draw conclusions based on the regression model
\[E(Y|X)=\beta_0+\beta_1 X\]
we need to know

- How the least squares parameter estimators vary from sample to sample, and
- how they deviate under the null hypothesis that there is no association between predictor and response
- Requires a statistical model

- Model the distribution of $Y$ given $X$ explicitly: f_{Y|X}(y)

## Modelling distribution of Y?

1. Besides *Linearity* we need additional assumptions!
2. *Independence*: Observations $(X_1,Y_1), ...,  (X_n,Y_n)$ are made for n independent subjects (is required to estimate the variance)
3. *Homoscedasticity* or *equal variances*: observations vary with equal mean around the regression line
    - Residuals $\epsilon_i$ have equal variance for each $X_i=x$
    - $\text{var}(Y\vert X=x) = \sigma^2$ for each $X=x$
    - $\sigma$ is referred to as the *residual standard deviation*
4. *Normality*: the residuals $\epsilon_i$ are normally distributed

![](https://raw.githubusercontent.com/GTPB/PSLS20/gh-pages/assets/figs/RegModel3.png){width=100%}


- Given 2, 3 and 4
$$\epsilon_i \text{ i.i.d.} N(0,\sigma^2).$$
- Together with 1 this implies:
$$Y_i\vert X_i\sim N(\beta_0+\beta_1 X_i,\sigma^2),$$

- We can show that given these assumption
$$\sigma^2_{\hat{\beta}_0}=\frac{\sum\limits_{i=1}^n X^2_i}{\sum\limits_{i=1}^n (X_i-\bar X)^2} \times\frac{\sigma^2}{n} \text{ en } \sigma^2_{\hat{\beta}_1}=\frac{\sigma^2}{\sum\limits_{i=1}^n (X_i-\bar X)^2}$$
- and the parameter estimators are also normally distributed
$$\hat\beta_0 \sim N\left(\beta_0,\sigma^2_{\hat \beta_0}\right) \text{ en } \hat\beta_1 \sim N\left(\beta_1,\sigma^2_{\hat \beta_1}\right)$$

## High spread of $X$ improves the precision

$$\sigma^2_{\hat{\beta}_1}=\frac{\sigma^2}{\sum\limits_{i=1}^n (X_i-\bar X)^2}$$

![](https://raw.githubusercontent.com/GTPB/PSLS20/gh-pages/assets/figs/spread.png){ width=100% }

- Conditional variance ($\sigma^2$) is unknown
- Estimate using *mean squared error* (MSE)
$$\hat\sigma^2=MSE=\frac{\sum\limits_{i=1}^n \left(y_i-\hat\beta_0-\hat\beta_1\times x_i\right)^2}{n-2}=\frac{\sum\limits_{i=1}^n e^2_i}{n-2}.$$
- This estimator is based on independence (assumption 2) and equality of the variance (assumption 3).
- Devide by $n-2$

Upon the estimation of $\sigma^2$ we obtain following standard errors:

$$\text{SE}_{\hat{\beta}_0}=\hat\sigma_{\hat{\beta}_0}=\sqrt{\frac{\sum\limits_{i=1}^n X^2_i}{\sum\limits_{i=1}^n (X_i-\bar X)^2} \times\frac{\text{MSE}}{n}} \text{ en } \text{SE}_{\hat{\beta}_1}=\hat\sigma_{\hat{\beta}_1}=\sqrt{\frac{\text{MSE}}{\sum\limits_{i=1}^n (X_i-\bar X)^2}}$$

- Again we can construct tests and confidence intervals using
$$T=\frac{\hat{\beta}_k-\beta_k}{SE(\hat{\beta}_k)} \text{ with } k=1,2.$$

- If all assumptions are valid $T$ follows t-verdeling with n-2 degrees of freedom.
\vspace{15pt}
- If no normality, but independence, linearity, equality of mean and large dataset
\[\rightarrow \text{Central Limit theorem}\]


### Breast cancer example

- Negative association between S100A8 and ESR1 gene expression.

- Generalize effect in sample to population using the confidence interval on the mean:
$$[\hat\beta_1 - t_{n-2,\alpha/2} \text{SE}_{\hat\beta_1},\hat\beta_1 + t_{n-2,\alpha/2} \text{SE}_{\hat\beta_1}]$$.

```{r}
confint(lm1)
```

- Negative association is significant on 5% significance level.


## Hypothesis test

- Translate the research question to assess the association between the S100A8 and ESR1 gene expression to parameters in the model.

- Under the null hypothesis of the absence of an association in the expression of both genes:
$$H_0: \beta_1=0$$

- Under the alternative hypothesis, there is an association between the expression of both genes :
$$H_1: \beta_1\neq0$$
\vspace{15pt}
- Test statistic
$$T=\frac{\hat{\beta}_1-0}{SE(\hat{\beta}_k)}$$
\vspace{15pt}
- Under $H_0$ the statistics follows a t-distribution with n-2 degrees of freedom.

### BRCA dataset

```{r}
summary(lm1)
```


- The association between the S100A8 and ESR1 expression is extremely significant  (p<<0.001).
- But, we first have to check all assumptions!
- Otherwise the conclusions based on the statistical test and the CI can be incorrect.

# Assess assumptions

- Independence: design
- Linearity: inference is useless if the association is not linear
- Homoscedasticity: inference/p-value is incorrect if data are heteroscedastic
- Normality: inference/p-value is incorrect if data are not normally distributed in small samples


## Linearity

```{r fig.align='center'}
brcaSubset %>%
  ggplot(aes(x=ESR1,y=S100A8)) +
  geom_point() +
  geom_smooth(se=FALSE,col="grey") +
  geom_smooth(method="lm",se=FALSE)
```


### Residual analysis

- Assumption of linearity is typically assessed using *residual plot*. (Especially if the lineair model has multiple covariates, later chapters)
- predictor of predictions $\hat\beta_0+\hat\beta_1 x$ on $X$-axis
- *residuals* on $Y$-as
$$e_i=y_i-\hat{g}(x_i)=y_i-\hat\beta_0-\hat\beta_1\times x_i,$$

```{r}
plot(lm1)
```

## Homoscedasticity (equal variances)

- Residuals and squared residuals cary information on the residual variability

- Association with predictors $\rightarrow$ indication of heteroscedasticity.
- Scatterplot of $e_i$ vs $x_i$ or predictions $\hat \beta_0+ \hat \beta_1 x_i$.
- Scatterplot van standardized residual versus $x_i$ or predictions.

## Normality

- If the sample size is large the estimators are normally distributed even if the observations are not normally distributed: central limit theorem
- How many observations? $\rightarrow$ depends on shape and magnitude of deviations
- Assumption: Data are Normally distributed conditional on X:
$$Y_i\vert X_i\sim N(\beta_0+\beta_1X_i,\sigma^2)$$
- QQ-plot of response Y is misleading and useless: distribution of $Y_i$ are different because they have a different conditional mean!

- QQ-plot of the residuals $e_i$

```{r echo=FALSE}
set.seed=200
par(mfrow=c(1,3))
x=rep(1:10,each=20)
y=x+rnorm(length(x))
boxplot(y~x)
qqnorm(y, main="Original observations")
qqline(y)
lmH<-lm(y~x)
plot(lmH,which=2,main="Residuals")
```


```{r}
plot(lm1,which=2)
```


# Invalid assumptions

- Transformation of predictor does not change distribution of Y for given X:

    - not useful to obtain homoscedasticity or Normal distribution
    - useful for linearity when normality and homoscedasticity are valid
    - Often inclusion of higher order terms: $X^2$, $X^3$, ...
    $$Y_i=\beta_0+\beta_1X_i+\beta_2X_i^2+ ... + \epsilon_i$$


- Transformation of response Y can be useful to obtain normality and homoscedasticity

-  $\sqrt(Y)$, $\log(Y)$, 1/Y, ...


## Breast cancer example

Problems with

- heteroscedasticity
- possibly deviations from normality (skewed to the right)
- negative concentration predictions are theoretically impossible
- non-linearity

This is often the case for concentration and intensity measurements

- These are often log-normal distributed (normal distribution upon log-transformatie)
- We also observed a kind of exponential relation with the smoother
- In gene expression literature often $\log_2$ transformation is adopted
- gene-expression on log scale: differences on log scale are fold changes on original scale!


```{r}
brca %>% ggplot(aes(x=ESR1,y=S100A8)) +
  geom_point() +
  geom_smooth()
```

```{r}
brca %>% ggplot(aes(x=ESR1%>%log2,y=S100A8%>%log2)) +
  geom_point() +
  geom_smooth()
```

```{r}
lm2<-lm(S100A8%>%log2 ~ ESR1 %>% log2, brca)
plot(lm2)
summary(lm2)
```


```{r}
confint(lm2)
```


### Interpretation 1

A patient with an ESR1 expression that is one unit on $\log_2$ scale higher than that of another patient on average has a $\log_2$ expression for S100A8 that is `r abs(round(lm2$coef[2],2))` units lower (95% CI [`r paste(round(confint(lm2)[2,],2),collapse=",")`]).

$$\log_2 \hat\mu_1=23.401  -1.615 \times \text{logESR}_1,\text{ } \log_2 \hat\mu_2=23.401  -1.615 \times \text{logESR}_2 $$
$$\log_2 \hat\mu_2-\log_2 \hat\mu_1=  -1.615 (\log_2 \text{ESR}_2-\log_2 \text{ESR}_1) = -1.615 \times 1 = -1.615$$

### Interpretation 2

Model on log-scale: upon back-transformation we obtain geometric means

\begin{eqnarray*}
\sum\limits_{i=1}^n \frac{\log x_i}{n}&=&\frac{\log x_1 + \ldots + \log x_n}{n}\\\\
&\stackrel{(1)}{=}&\frac{\log(x_1 \times \ldots \times x_n)}{n}=\frac{\log\left(\prod\limits_{i=1}^n x_i\right)}{n}\\\\
&\stackrel{(2)}{=}&\log \left(\sqrt[\leftroot{-1}\uproot{2}\scriptstyle n]{\prod\limits_{i=1}^n x_i}\right)
\end{eqnarray*}

- Population mean $\mu$ is estimated as a geometric mean
- Logarithmic transformation is monotone: we can backtransform confidence intervals on log-scale!


```{r}
2^lm2$coef[2]
2^-lm2$coef[2]
2^-confint(lm2)[2,]
```

A patient with an ESR1 expression that is 2 times the expression of that of another patient will on average have an  S100A8 expression that is `r round(2^-lm2$coef[2]
,2)` times lower (95\% CI [`r paste(sort(round(2^-confint(lm2)[2,],2)),collapse=",")`]).


$$\log_2 \hat\mu_1=23.401  -1.615 \times \text{logESR}_1,\text{ } \log_2 \hat\mu_2=23.401  -1.615 \times \text{logESR}_2 $$
$$\log_2 \hat\mu_2-\log_2 \hat\mu_1=  -1.615 (\log_2 \text{ESR}_2-\log_2 \text{ESR}_1) $$
$$\log_2 \left[\frac{\hat\mu_2}{\hat\mu_1}\right]=  -1.615 \log_2\left[\frac{ \text{ESR}_2}{\text{ESR}_1}\right] $$
$$\frac{\hat\mu_2}{\hat\mu_1}=\left[\frac{ \text{ESR}_2}{\text{ESR}_1}\right]^{-1.615}=2^ {-1.615} =0.326$$
or
$$\frac{\hat\mu_1}{\hat\mu_2}=2^{1.615} =3.06$$


### Interpretation 3

A patient with an ESR1 expression that is 1\% higher than that of another patient will on average have  an expression-level for S100A8 gen  that is approximately `r round(lm2$coef[2],2)`% lower (95\% CI [`r paste(round(confint(lm2)[2,],2),collapse=",")`])%.

$$\log_2 \hat\mu_1=23.401  -1.615 \times \text{logESR}_1,\text{ } \log_2 \hat\mu_2=23.401  -1.615 \times \text{logESR}_2 $$
$$\log_2 \hat\mu_2-\hat\log_2 \mu_1=  -1.615 (\log_2 \text{ESR}_2-\log_2 \text{ESR}_1) $$
$$\log_2 \left[\frac{\hat\mu_2}{\hat\mu_1}\right]=  -1.615 \log_2\left[\frac{ \text{ESR}_2}{\text{ESR}_1}\right] $$
$$\frac{\hat\mu_2}{\hat\mu_1}=\left[\frac{ \text{ESR}_2}{\text{ESR}_1}\right]^{-1.615}=1.01^ {-1.615} =0.984 \approx -1.6\%$$

This is valid for low to moderate values of $\beta_1$:
$$-10<\beta_1<10 \rightarrow 1.01^{\beta_1} -1 \approx \frac{\beta_1}{100}.$$


## Inference on the mean outcome

- A regression model can also be used for prediction
- Inference on average outcome for a given value of $X=x$, i.e.
$$\hat{g}(x)= \hat{\beta}_0 + \hat{\beta}_1 x$$
- $\hat{g}(x)$ is an estimator of the conditional mean $E[Y\vert X=x]$
- Parameter estimators are Normally distributed and unbiased $\rightarrow$ estimator $\hat{g}(x)$ is also Normally distributed and unbiased.

$$\text{SE}_{\hat{g}(x)}=\sqrt{MSE\left\{\frac{1}{n}+\frac{(x-\bar X)^2}{\sum\limits_{i=1}^n (X_i-\bar X)^2}\right\}}.$$

$$T=\frac{\hat{g}(x)-g(x)}{SE_{\hat{g}(x)}}\sim t_{n-2}$$

- Mean response and confidence intervals for the mean response in R via de `predict(.)` functie.
- `newdata` argument: predictor values (x-values) at which we want to calculate the mean response
- `interval="confidence"` argument to obtain CI.
- Without newdata argument we perform predictions for all predictor values in the dataset used to fit the model.

```{r}
grid <- 140:4000
g <- predict(lm2,newdata=data.frame(ESR1=grid), interval="confidence")
head(g)
```

Note, that we do not have to transform the new data that we specified for the ESR1 expression because we fitted the model with a call to the `lm` function and specified the transformation within the lm formula using the pipe command!

```{r}
brca %>% ggplot(aes(x=ESR1%>%log2,y=S100A8%>%log2)) +
  geom_point() +
  geom_smooth(method="lm")
```

## Back-transformation
```{r}
newdata<-data.frame(cbind(grid,2^g))
brca %>% ggplot(aes(x=ESR1,y=S100A8)) +
  geom_point() +
  geom_line(aes(x=grid,y=fit),newdata) +
  geom_line(aes(x=grid,y=lwr),newdata,color="grey") +
  geom_line(aes(x=grid,y=upr),newdata,color="grey")
```

# Prediction-intervals

- We can also make a prediction for the location of a new observation that would be collected in a new experiment for a patient with a particular value for their ESR1 expression

- It is important to notice that this experiment still has to be conducted. So we want to predict the non-observed individual expression value for a novel patient.

- For a novel independent observation $Y^*$
$$
  Y^* = g(x) + \epsilon^*
$$
with $\epsilon^*\sim N(0,\sigma^2)$ and $\epsilon^*$ independent of the observations in the sample $Y_1,\ldots, Y_n$.

- We predict a new log-S100A8 for a patient with a known log2-ESR1 expression level x
\[
  \hat{y}(x)=\hat{\beta}_0+\hat{\beta}_1 \times x
\]

- The estimated mean outcome and prediction for a new observation are equal.

- But, their sample distributions are different!

    - Uncertainty on the estimated mean outcome  $\leftarrow$ uncertainty on estimated model parameters $\hat\beta_0$ en $\hat\beta_1$.
    - Uncertainty on new observation $ $\leftarrow$ *uncertainty on estimated mean* and  *additional uncertainty* because the new observation will deviate around the mean!


$$\text{SE}_{\hat{Y}(x)}=\sqrt{\hat\sigma^2+\hat\sigma^2_{\hat{g}(x)}}=\sqrt{MSE\left\{1+\frac{1}{n}+\frac{(x-\bar X)^2}{\sum\limits_{i=1}^n (X_i-\bar X)^2}\right\}}.$$

$$\frac{\hat{Y}(x)-Y}{\text{SE}_{\hat{Y}(x)}}\sim t_{n-2}$$

- Note, that a **prediction-interval** (PI) is an improved version of a reference-interval when the model parameters are unknown: Uncertainty on model parameters +  t-distribution.


```{r}
p <- predict(lm2,newdata=data.frame(ESR1=grid), interval="prediction")
head(p)
```

```{r}
preddata<-data.frame(cbind(grid=grid%>%log2,p))
brca %>% ggplot(aes(x=ESR1%>%log2,y=S100A8%>%log2)) +
  geom_point() +
  geom_smooth(method="lm") +
     geom_line(aes(x=grid,y=lwr),preddata,color="blue") +
  geom_line(aes(x=grid,y=upr),preddata,color="blue")
```

```{r}
preddata<-data.frame(cbind(grid,2^p))
brca %>% ggplot(aes(x=ESR1,y=S100A8)) +
  geom_point() +
  geom_line(aes(x=grid,y=fit),newdata) +
  geom_line(aes(x=grid,y=lwr),newdata,color="grey") +
  geom_line(aes(x=grid,y=upr),newdata,color="grey") +
    geom_line(aes(x=grid,y=lwr),preddata,color="blue") +
  geom_line(aes(x=grid,y=upr),preddata,color="blue")
```


## NHANES example


- Replace reference interval for cholesterol level from chapter 2 by prediction-interval.

- Reference interval

```{r}
library(NHANES)
fem <- NHANES %>% filter(Gender=="female"&!is.na(DirectChol))

exp(fem$DirectChol%>%log%>%mean + c(-1,1)* qnorm(0.975) * (fem$DirectChol%>%log%>%sd))
```

- prediction interval

```{r}
lmChol <- lm(DirectChol %>% log2~1,data=fem)
predInt <- predict(lmChol,interval="prediction",newdata=data.frame(noPred=1))
round(2^predInt,2)
```


Note, that the prediction interval is almost similar to the reference interval for the large sample. Indeed we could estimate the parameters very precise.

We will do the same thing for the small sample size of 10 patients.

- Reference interval

```{r}
set.seed(1)
fem10<- NHANES %>% filter(Gender=="female"&!is.na(DirectChol)) %>% sample_n(size=10)

2^(fem10$DirectChol%>%log2%>%mean + c(-1,1)* qnorm(0.975) * (fem10$DirectChol%>%log2%>%sd))
```

- Prediction interval

```{r}
lmChol10 <- lm(DirectChol %>% log2~1,data=fem10)
predInt10 <- predict(lmChol10,interval="prediction",newdata=data.frame(noPred=1))
round(2^predInt10,2)
```

- Note, that the PI now captures uncertainty in parameter estimators (mean and standard error).
And that the interval becomes much wider! This is particularly important here for the upper limit because we back-transformed the data!

- The interval is almost as wide as the one based on the large sample.

- In small samples it is very important to account for this additional uncertainty.


# Sum of squares and Anova-table
##Total sum of squares
$$\text{SSTot} = \sum_{i=1}^n (Y_i-\bar{Y})^2.$$

- SStot can be used to estimate the variance of the **marginal distribution** of the response.

- In this chapter we focused on the **conditional distribution** $f(Y\vert X=x)$.

- We known that MSE is a good estimate of the variance of the conditional distribution of  $Y\vert X=x$.


```{r out.width='100%', fig.asp=.8, fig.align='center', echo=FALSE}
brca$log2ESR1<-log2(brca$ESR1)
brca$log2S100A8<-log2(brca$S100A8)
plot(log2S100A8~log2ESR1,data=brca,xlab="ESR1 expressie (log2)",ylab="S100A8 expressie (log2)",cex.axis=1.5,cex.main=1.5,cex.lab=1.5,col=4)
abline(h=mean(brca$log2S100A8))
for (i in 1:length(brca$log2S100A8)) lines(rep(brca$log2ESR1[i],2),c(mean(brca$log2S100A8),brca$log2S100A8[i]),lty=2,col=4)
```

## Sum of squares of the regression SSR

$$\text{SSR} = \sum_{i=1}^n (\hat{Y}_i - \bar{Y})^2 = \sum_{i=1}^n (\hat{g}(x_i) - \bar{Y})^2.$$

- Is a measure for the deviation of the predictions on the regression line and the marginal mean of the response.

- Another interpretation: difference between two models

    - Estimated model $\hat{g}(x)=\hat\beta_0+\hat\beta_1x$
    - Estimated model without predictor (only intercept): $g(x)=\beta_0$ $\rightarrow$ $\beta_0$ will be equal to $\bar{Y}$.

- SSR measures the size of the effect of the predictor

```{r out.width='100%', fig.asp=.8, fig.align='center',echo=FALSE}
plot(log2S100A8~log2ESR1,brca,xlab="ESR1 expressie (log2)",ylab="S100A8 expressie (log2)",cex.axis=1.5,cex.main=1.5,cex.lab=1.5)
abline(h=mean(brca$log2S100A8))
abline(lm2,col=2)
points(brca$log2ESR1,lm2$fitted,pch=2,col=2)
for (i in 1:length(brca$log2S100A8)) lines(rep(brca$log2ESR1[i],2),c(mean(brca$log2S100A8),lm2$fitted[i]),lty=2,col=2)
```


## Sum of Squares of the Error

$$ \text{SSE} = \sum_{i=1}^n (Y_i-\hat{Y}_i )^2 = \sum_{i=1}^n \left\{Y_i-\hat{g}\left(x_i\right)\right\}^2.$$

- The smaller SSE the better the fit.


- Least squares method!

---

```{r out.width='100%', fig.asp=.8, fig.align='center',echo=FALSE}
plot(log2S100A8~log2ESR1,brca,xlab="ESR1 expressie (log2)",ylab="S100A8 expressie (log2)",cex.axis=1.5,cex.main=1.5,cex.lab=1.5)
abline(lm2,col=2)
points(brca$log2ESR1,lm2$fitted,pch=2,col=2)
for (i in 1:length(brca$log2S100A8)) lines(rep(brca$log2ESR1[i],2),c(brca$log2S100A8[i],lm2$fitted[i]),lty=2)
```

We can show that SST can be decomposed in
\begin{eqnarray*}
  \text{SSTot}
    &=&  \sum_{i=1}^n (Y_i-\bar{Y})^2 \\
    &=&  \sum_{i=1}^n (Y_i-\hat{Y}_i+\hat{Y}_i-\bar{Y})^2 \\
    &=&  \sum_{i=1}^n (Y_i-\hat{Y}_i)^2+\sum_{i=1}^n(\hat{Y}_i-\bar{Y})^2 \\
    &=&  \text{SSE }+\text{SSR}
  \end{eqnarray*}

-  Total variability in the data (SSTot) is partially explained by the predictor (SSR).
- Variability that we cannot explain with the regression model is the residual variability (SSE).


## Determination coefficient

$$ R^2 = 1-\frac{\text{SSE}}{\text{SSTot}}=\frac{\text{SSR}}{\text{SSTot}}.$$

- *Fraction of total variability of the sample outcomes explained by the model*.

- Large $R^2$ indicates that the model has the potential to make good predictions  (small SSE).

- Not very indicative for p-value of the test $H_0:\beta_1=0$ vs $H_1:\beta_1\neq0$.

  - p-value is largely determined by SSE and sample size $n$, but not by SSTot.
  - $R^2$ is determined by SSE and SSTot but not by sample size $n$.
- Model with low $R^2$ is still useful to study associations as long as the association is modelled correctly!

### Breast cancer example

```{r}
summary(lm2)
```

## F-Test in simple linear model

- Sum of squares are the bases for $F$-tests
$$  F  = \frac{\text{MSR}}{\text{MSE}}$$

with  $\text{MSR} = \frac{\text{SSR}}{1} \text{ and } \text{MSE} = \frac{\text{SSE}}{n-2}.$

- MSR mean sum of squares of the regression,

- denominators 1 en $n-2$ are the degrees of freedom of SSR and SSE.

- Under $H_0: \beta_1=0$
$$H_0:F = \frac{\text{MSR}}{\text{MSE}} \sim F_{1,n-2},$$
- F-test is always two-sided! $H_1:\beta_1\neq 0$
$$  p = P_0\left[F\geq f\right]=1-F_F(f;1,n-2)$$


```{r}
summary(lm2)
```


```{r, echo=FALSE}
grid<-seq(0,10,.1)
plot(grid,df(grid,1,30),type="l",xlab="F",ylab="Density",main="F-distribution with 1 df in the nominator and 30 in the denominator",cex.main=1.5,cex.axis=1.5,cex.lab=1.5)
```


## Anova Table


| |Df|Sum Sq|Mean Sq|F value|Pr(>F)|
|---|---|---|---|---|---|
|Regression|degrees of freedom SSR|SSR|MSR|f-statistic|p-value|
|Error|degrees of freedom SSE|SSE|MSE| | |

```{r}
anova(lm2)
```


# Dummy variables

- Linear regression model  can also be used to compare two group means.
- brca: difference in average age between patients with unaffected and affected lymph nodes.

- Define dummy variabele
$$x_i = \left\{ \begin{array}{ll}
1 & \text{affected lymph nodes} \\
0 & \text{unaffected lymph nodes} \end{array}\right.$$

- group with $x_i=0$ is referred to as the  **reference group**.

- Regression model remains unaltered,
$$Y_i = \beta_0 + \beta_1 x_i +\epsilon_i$$
with $\epsilon_i \text{ iid } N(0,\sigma^2)$


Because $x_i$ only can take two values, we can study the regression model for each value of  $x_i$ separately:
$$ \begin{array}{lcll}
   Y_i &=& \beta_0 +\epsilon_i &\text{unaffected lymph nodes} (x_i=0) \\
   Y_i &=& \beta_0 + \beta_1 +\epsilon_i &\text{ affected lymph nodes} (x_i=1) .
 \end{array}$$
So
 \begin{eqnarray*}
   E\left[Y_i\mid x_i=0\right] &=& \beta_0 \\
   E\left[Y_i\mid x_i=1\right] &=& \beta_0 + \beta_1,
\end{eqnarray*}

 Hence, the interpretation of $\beta_1$:
$$   \beta_1 = E\left[Y_i\mid x_i=1\right]-E\left[Y_i\mid x_i=0\right]$$

$\beta_1$ is the average age difference between patients with affected and patients with unaffected lymph nodes (reference group).

With notation $\mu_0= E\left[Y_i\mid x_i=0\right]$ and $\mu_1= E\left[Y_i\mid x_i=1\right]$ this becomes
$$\beta_1 = \mu_1-\mu_0.$$

We can show that
$$\begin{array}{ccll}
 \hat\beta_0
   &=& \bar{Y}_1&\text{ (sample mean of reference group)} \\
 \hat\beta_1
   &=& \bar{Y}_2-\bar{Y}_1&\text{(estimator of effect size)} \\
 \text{MSE}
   &=& S_p^2 .
\end{array}$$

Tests $H_0:\beta_1=0$ vs.  $H_1:\beta_1\neq0$ can be used to assess the null hypothesis of the  two-sample $t$-test, $H_0:\mu_1=\mu_2$ vs $H_1:\mu_1\neq\mu_2$.


```{r}
brca$node <- as.factor(brca$node)
t.test(age~node,brca,var.equal=TRUE)
```

```{r}
lm3 <- lm(age~node,brca)
summary(lm3)
```

```{r}
plot(lm3)
```



```{r}
brca %>% ggplot(aes(x=node%>%as.factor,y=age)) +
  geom_boxplot()
```


```{r out.width='100%', fig.asp=.8, fig.align='center'}
par(mfrow=c(3,3))
set.seed(354)
for(i in 1:9) plot(rnorm(32)~node,brca,ylab="iid N(0,1)")
```


# Observational study

- We cannot conclude that age causes a higher risk for affected lymph nodes.
- Possibly **confounding**: no randomisation $\rightarrow$ groups of patients with affected and unaffected lymph nodes. They can also differ in other characteristics.

- We can only conclude that there is an association between lymph node status and age.

- However, the association does not have to be causal!


- Note, that this is also the case for the linear model for $\log_2$-S100A8-expression.

    - Because we were not able to fix the  ESR1-expression experimentally we cannot conclude that a higher ESR1-expression causes a decrease in the S100A8-expression.
    - We can only conclude that there is a negative association.
    - To assess the impact of a gene on other gene typically knockout mutants are used in the lab.
</div>
<div class="footer">
    <hr>
    This work is licensed under the <a href= "https://creativecommons.org/licenses/by-nc-sa/4.0">
    CC BY-NC-SA 4.0</a> licence.
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("06-linearRegression.Rmd");
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
