<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement and Jeroen Gilis" />


<title>Exercises on chapter 8: Multiple linear regression</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>





<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>


<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PSLS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01-intro.html">1. Introduction</a>
    </li>
    <li>
      <a href="02-concepts.html">2. Concepts</a>
    </li>
    <li>
      <a href="03-experimentalDesign.html">3. Experimental Design I</a>
    </li>
    <li>
      <a href="04-dataExploration.html">4. Data Exploration</a>
    </li>
    <li>
      <a href="05-statisticalInference.html">5.1 Statistical Inference 1</a>
    </li>
    <li>
      <a href="05-statisticalInference-twosampleT.html">5.2 Statistical Inference 2</a>
    </li>
    <li>
      <a href="06-linearRegression.html">6. Linear Regression</a>
    </li>
    <li>
      <a href="07-Anova.html">7. Analysis of Variance</a>
    </li>
    <li>
      <a href="08-MultipleRegression.html">8. Multiple Regression</a>
    </li>
    <li>
      <a href="08-MultipleRegression-SpecificDesigns.html">8.2 Multiple Regression: Factorial Designs</a>
    </li>
    <li>
      <a href="08-experimentalDesignII.html">8.3. Experimental Design II</a>
    </li>
    <li>
      <a href="08-experimentalDesignIII.html">8.4. Experimental Design II</a>
    </li>
    <li>
      <a href="09-NonparametericStatistics-WilcoxonMannWithney.html">9.1. Nonparametric Statistics 1</a>
    </li>
    <li>
      <a href="09-NonparametericStatistics-KruskalWallis.html">9.2. Nonparametric Statistics 2</a>
    </li>
    <li>
      <a href="10-categoricalDataAnalysis.html">10. Categorical Data Analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    Labs
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="04_exercises.html">4. Data Exploration</a>
    </li>
    <li>
      <a href="05_exercises.html">5. Statistical inference</a>
    </li>
    <li>
      <a href="06_exercises.html">6. Simple linear regression</a>
    </li>
    <li>
      <a href="07_exercises.html">7. ANOVA</a>
    </li>
    <li>
      <a href="08_exercises.html">8. Multiple linear regression</a>
    </li>
    <li>
      <a href="09_exercises.html">9. Non-parametric statistics</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/statOmics/PSLS">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="http://statomics.github.io/">statOmics</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Exercises on chapter 8: Multiple linear
regression</h1>
<h4 class="author">Lieven Clement and Jeroen Gilis</h4>
<h4 class="date">statOmics, Ghent University (<a
href="https://statomics.github.io"
class="uri">https://statomics.github.io</a>)</h4>

</div>


<p>On the fourth day of the “Practical Statistics for the Life Sciences
(2021)” course, we will have four tutorials on multiple linear
regression, based on different datasets:</p>
<div id="the-poison-dataset" class="section level1" number="1">
<h1><span class="header-section-number">1</span> The poison dataset</h1>
<p>In this experiment, 96 fish (dojofish, goldfish and zebrafish) were
placed separately in a tank with two liters of water and a certain dose
(in mg) of the poison EI-43,064. The resistance of the fish against the
poison was measured as the amount of minutes the fish survived after
being exposed to the poison (<code>Surv_time</code>, in minutes).
Additionally, the weight of each fish was measured.</p>
<p>In this tutorial session we will focus on Dojofish, and we will model
the survival time in function of the poison dose while correcting for
the weight of the fish.</p>
<ol style="list-style-type: decimal">
<li><p>We will first analyse the survival data by only considering the
dose as an explanatory variable for survival time</p></li>
<li><p>Next we will model the survival data with and <strong>additive
model</strong> for dose and weight</p></li>
</ol>
<ul>
<li>Exercise: <a href="./08_1_poison1.html">Exercise1</a></li>
<li>Data path: “<a
href="https://raw.githubusercontent.com/statOmics/PSLS21/data/poison.csv"
class="uri">https://raw.githubusercontent.com/statOmics/PSLS21/data/poison.csv</a>”
<!-- - Solution1: [Solution1](./08_1_poison1_sol.html) --></li>
</ul>
<hr />
</div>
<div id="the-poison-dataset-2" class="section level1" number="2">
<h1><span class="header-section-number">2</span> The poison dataset
(2)</h1>
<p>Again, we will work with the poison dataset. In contrast to the
exercise above, we here we will model the survival time in function of
the dose and the weight of the fish, <strong>and including an
interaction between dose and weight.</strong></p>
<ul>
<li>Exercise: <a href="./08_2_poison2.html">Exercise2</a></li>
<li>Data path: “<a
href="https://raw.githubusercontent.com/statOmics/PSLS21/data/poison.csv"
class="uri">https://raw.githubusercontent.com/statOmics/PSLS21/data/poison.csv</a>”
<!-- - Solution2: [Solution2](./08_2_poison1_sol.html) --></li>
</ul>
<hr />
</div>
<div id="the-puromycin-dataset" class="section level1" number="3">
<h1><span class="header-section-number">3</span> The puromycin
dataset</h1>
<p>Data on the velocity of an enzymatic reaction were obtained by
Treloar (1974). The number of counts per minute of radioactive product
from the reaction was measured as a function of substrate concentration
in parts per million (ppm) and from these counts the initial rate (or
velocity) of the reaction was calculated (counts/min/min). The
experiment was conducted once with the enzyme treated with Puromycin,
and once with the enzyme untreated.</p>
<p>Assess if there is an association between the substrate concentration
and rate <strong>for both the treated and untreated enzymes.</strong> To
do this, fit a model that includes a main effect for concentration, a
main effect for enzyme state, and an interaction term between these two
variables.</p>
<ul>
<li>Exercise: <a href="./08_3_puromycin.html">Exercise3</a></li>
<li>Data path: Not required
<!-- - Solution: [Solution3](./08_3_puromycin_sol.html) --></li>
</ul>
<hr />
</div>
<div id="the-kpna2-dataset" class="section level1" number="4">
<h1><span class="header-section-number">4</span> The KPNA2 dataset</h1>
<p>Histologic grade in breast cancer provides clinically important
prognostic information. Researchers examined whether histologic grade
was associated with gene expression profiles of breast cancers and
whether such profiles could be used to improve histologic grading.</p>
<p>In this tutorial we will assess the association between histologic
grade and the expression of the KPNA2 gene that is known to be
associated with poor breast cancer prognosis. The patients, however, do
not only differ in the histologic grade, but also on their lymph node
status. The lymph nodes were not affected (0) or surgically removed
(1).</p>
<ul>
<li>Exercise: <a href="./08_4_kpna2.html">Exercise4</a></li>
<li>Data path: “<a
href="https://raw.githubusercontent.com/statOmics/SGA21/master/data/kpna2.txt"
class="uri">https://raw.githubusercontent.com/statOmics/SGA21/master/data/kpna2.txt</a>”
<!-- - Solution: [Solution4](./08_4_kpna2_sol.html) --></li>
</ul>
<hr />
</div>
<div id="blocking-the-rats-dataset-1" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Blocking: the rats
dataset (1)</h1>
<p>Researchers are studying the impact of protein sources and protein
levels in the diet on the weight of rats. They feed the rats with diets
of beef, cereal and pork and use a low and high protein level for each
diet type. The researchers can include 60 rats in the experiment. Prior
to the experiment, the rats were divided in 10 homogeneous groups of 6
rats based on characteristics such as initial weight, appetite, etc.</p>
<p>Within each group a rat is randomly assigned to a diet. The rats are
fed during a month and the weight gain in grams is recorded for each
rat.</p>
<p>The researchers want to assess the effect of the type of diet and the
protein level on the weight of the rats.</p>
<p>In this exercise we will perform the data exploration using all
diets, but, to keep the data analysis simple we will only assess the
beef and cereal diets.</p>
<ul>
<li>Exercise: <a href="./08_5_rats.html">Exercise5</a></li>
<li>Data path: “<a
href="https://raw.githubusercontent.com/statOmics/PSLS21/data/dietRats.txt"
class="uri">https://raw.githubusercontent.com/statOmics/PSLS21/data/dietRats.txt</a>”
<!-- - Exercise: [Solution5](./08_5_rats_sol.html) --></li>
</ul>
<hr />
</div>
<div id="blocking-the-rats-dataset-2" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Blocking: the rats
dataset (2)</h1>
<p>Again, we make use of the <em>rats</em> dataset defined above. In
contrast to the previous exercise, we perform the analysis for all three
diets in the dataset.</p>
<ul>
<li>Exercise: <a href="./08_6_rats.html">Exercise6</a></li>
<li>Data path: “<a
href="https://raw.githubusercontent.com/statOmics/PSLS21/data/dietRats.txt"
class="uri">https://raw.githubusercontent.com/statOmics/PSLS21/data/dietRats.txt</a>”
<!-- - Exercise: [Solution6](./08_6_rats_sol.html) --></li>
</ul>
<hr />
</div>
<div id="power-analysis-the-rodents-dataset" class="section level1"
number="7">
<h1><span class="header-section-number">7</span> Power analysis: the
rodents dataset</h1>
<p>A biologist examined the effect of a fungal infection on the eating
behavior of rodents. Infected apples were offered to a group of eight
rodents, and sterile apples were offered to a group of 4 rodents. The
amount of grams of apples consumed per kg body weight are given in the
dataset.</p>
<p>We will answer four research questions:</p>
<ul>
<li><p>What is the power of the experiment if the effect size and
standard deviation in the population would be equal to the ones you
observed in the experiment?</p></li>
<li><p>What would the power by if number of rodents would balanced in
both groups?</p></li>
<li><p>How many observations would you need to pick up the treatment
effect with a power of 90%?</p></li>
<li><p>Suppose that we would like to pick up an effect size of <span
class="math inline">\(\beta_1 = 60 g/kg\)</span>. How many samples would
be required in each group to obtain a power of 90%?</p></li>
<li><p>Exercise: <a
href="./08_ExperimentalDesignII_1_rodents.html">Exercise7</a></p></li>
<li><p>Data path: Not required
<!-- - Solution: [Solution7](./08_ExperimentalDesignII_1_rodents_sol.html) --></p></li>
</ul>
<hr />
</div>
<div id="power-analysis-the-puromycin-dataset" class="section level1"
number="8">
<h1><span class="header-section-number">8</span> Power analysis: the
Puromycin dataset</h1>
<p>Data on the velocity of an enzymatic reaction were obtained by
Treloar (1974). The number of counts per minute of radioactive product
from the reaction was measured as a function of substrate concentration
in parts per million (ppm) and from these counts the initial rate (or
velocity) of the reaction was c alculated (counts/min/min). The
experiment was conducted once with the enzyme treated with Puromycin,
and once with the enzyme untreated.</p>
<p>We will answer four research questions:</p>
<ul>
<li><p>Use the data to calculate the power to pick up an association
that is as least as strong as the association you observed in the
dataset when using an experiment with the same design.</p></li>
<li><p>Use the data to calculate the power to pick up an association
where the reaction rate increases on average with 10 counts/min when the
substrate concentration is 10 times higher (<span
class="math inline">\(\beta_1=10\)</span>).</p></li>
<li><p>Use the data to calculate the number of repeats you need for each
concentration to pick up an association where the reaction rate
increases on average with 10 counts/min when the substrate concentration
is 10 times higher with a power of at least 90%. (<span
class="math inline">\(\beta_1=10\)</span>)</p></li>
<li><p>Suppose that you would setup an experiment with a design similar
with the same concentrations as in the puromycin dataset and you have
the following restriction: you need to use each concentration at least
once and can setup at most 12 reactions, how would you choose your
design points? Calculate the power for this design when the effect size
is 10 counts/min per 10 times increase in the substrate concentration
(<span class="math inline">\(\beta_1=10\)</span>).</p></li>
<li><p>Exercise: <a
href="./08_ExperimentalDesignII_1_puromycin.html">Exercise8</a></p></li>
<li><p>Data path: Not required
<!-- - Solution: [Solution8](./08_ExperimentalDesignII_1_puromycin_sol.html) --></p></li>
</ul>
<hr />
</div>
<div id="power-analysis-he-kpna2-dataset" class="section level1"
number="9">
<h1><span class="header-section-number">9</span> Power analysis: he
KPNA2 dataset</h1>
<p>Histologic grade in breast cancer provides clinically important
prognostic information. Researchers examined whether histologic grade
was associated with gene expression profiles of breast cancers and
whether such profiles could be used to improve histologic grading.</p>
<p>In this tutorial we will assess the association between histologic
grade and the expression of the KPNA2 gene that is known to be
associated with poor breast cancer prognosis. The patients, however, do
not only differ in the histologic grade, but also on their lymph node
status. The lymph nodes were not affected (0) or surgically removed
(1).</p>
<p>We will answer three research questions:</p>
<ul>
<li><p>What is the power to pick up each of the contrasts when their
real effect sizes would be equal to the effect sizes we observed in the
study?</p></li>
<li><p>How does the power evolves if we have 2 upto 10 repeats for each
factor combination of grade and node when their real effect sizes would
be equal to the ones we observed in the study?</p></li>
<li><p>What is the power to pick up each of the contrasts when the FC
for grade for patients with unaffected lymph nodes equals 1.5 (<span
class="math inline">\(\beta_g = log2(1.5)\)</span>)?</p></li>
<li><p>Data path: “<a
href="https://raw.githubusercontent.com/statOmics/SGA21/master/data/kpna2.txt"
class="uri">https://raw.githubusercontent.com/statOmics/SGA21/master/data/kpna2.txt</a>”
<!-- - Solution: [Solution9](./08_ExperimentalDesignII_3_kpna2_sol.Rmd.html) --></p></li>
</ul>
</div>

<div id="rmd-source-code">---
title: "Exercises on chapter 8: Multiple linear regression"
author: "Lieven Clement and Jeroen Gilis"
date: "statOmics, Ghent University (https://statomics.github.io)"
---

On the fourth day of the "Practical Statistics for the Life Sciences (2021)"
course, we will have four tutorials on multiple linear regression, based on
different datasets:


# The poison dataset

In this experiment, 96 fish (dojofish, goldfish and zebrafish)
were placed separately in a tank with two liters of water and
a certain dose (in mg) of the poison EI-43,064. The resistance
of the fish against the poison was measured as the amount of
minutes the fish survived after being exposed to the poison (`Surv_time`, in
minutes). Additionally, the weight of each fish was measured.

In this tutorial session we will focus on Dojofish, and we will model the
survival time in function of the poison dose while correcting for the weight of
the fish.

1. We will first analyse the survival data by only considering the dose as an
explanatory variable for survival time

2. Next we will model the survival data with and **additive model** for dose and
weight

- Exercise: [Exercise1](./08_1_poison1.html)
- Data path: "https://raw.githubusercontent.com/statOmics/PSLS21/data/poison.csv"
<!-- - Solution1: [Solution1](./08_1_poison1_sol.html) -->

---

# The poison dataset (2)

Again, we will work with the poison dataset. In contrast to the exercise
above, we here we will model the survival time in function of the dose and the
weight of the fish, **and including an interaction between dose and weight.**

- Exercise: [Exercise2](./08_2_poison2.html)
- Data path: "https://raw.githubusercontent.com/statOmics/PSLS21/data/poison.csv"
<!-- - Solution2: [Solution2](./08_2_poison1_sol.html) -->

---

# The puromycin dataset

Data on the velocity of an enzymatic reaction were obtained by Treloar (1974).
The number of counts per minute of radioactive product from the reaction was
measured as a function of substrate concentration in parts per million (ppm) and
from these counts the initial rate (or velocity) of the reaction was calculated (counts/min/min). The experiment was conducted once with the enzyme treated
with Puromycin, and once with the enzyme untreated.

Assess if there is an association between the substrate concentration and rate
**for both the treated and untreated enzymes.** To do this, fit a model that
includes a main effect for concentration, a main effect for enzyme state, and
an interaction term between these two variables.

- Exercise: [Exercise3](./08_3_puromycin.html)
- Data path: Not required
<!-- - Solution: [Solution3](./08_3_puromycin_sol.html) -->

---

# The KPNA2 dataset

Histologic grade in breast cancer provides clinically important prognostic
information. Researchers examined whether histologic grade was associated with
gene expression profiles of breast cancers and whether such profiles could be
used to improve histologic grading.

In this tutorial we will assess the association between histologic grade and
the expression of the KPNA2 gene that is known to be associated with poor breast
cancer prognosis. The patients, however, do not only differ in the histologic
grade, but also on their lymph node status. The lymph nodes were
not affected (0) or surgically removed (1).

- Exercise: [Exercise4](./08_4_kpna2.html)
- Data path: "https://raw.githubusercontent.com/statOmics/SGA21/master/data/kpna2.txt"
<!-- - Solution: [Solution4](./08_4_kpna2_sol.html) -->

---

# Blocking: the rats dataset (1)

Researchers are studying the impact of protein sources and protein levels in
the diet on the weight of rats. They feed the rats with diets of beef, cereal
and pork and use a low and high protein level for each diet type.
The researchers can include 60 rats in the experiment. Prior to the experiment,
the rats were divided in 10 homogeneous groups of 6 rats based on
characteristics such as initial weight, appetite, etc.

Within each group a rat is randomly assigned to a diet. The rats are fed during
a month and the weight gain in grams is recorded for each rat.

The researchers want to assess the effect of the type of diet and the protein
level on the weight of the rats.

In this exercise we will perform the data exploration using all diets, but,
to keep the data analysis simple we will only assess the beef and cereal diets.

- Exercise: [Exercise5](./08_5_rats.html)
- Data path: "https://raw.githubusercontent.com/statOmics/PSLS21/data/dietRats.txt"
<!-- - Exercise: [Solution5](./08_5_rats_sol.html) -->

---

# Blocking: the rats dataset (2)

Again, we make use of the *rats* dataset defined above. In contrast to the
previous exercise, we perform the analysis for all three diets in the dataset.

- Exercise: [Exercise6](./08_6_rats.html)
- Data path: "https://raw.githubusercontent.com/statOmics/PSLS21/data/dietRats.txt"
<!-- - Exercise: [Solution6](./08_6_rats_sol.html) -->


---

# Power analysis: the rodents dataset

A biologist examined the effect of a fungal infection on the eating behavior
of rodents. Infected apples were offered to a group of eight rodents,
and sterile apples were offered to a group of 4 rodents. The amount of grams
of apples consumed per kg body weight are given in the dataset.

We will answer four research questions:

- What is the power of the experiment if the effect size and standard deviation
in the population would be equal to the ones you observed in the experiment?

- What would the power by if number of rodents would  balanced in both groups?

- How many observations would you need to pick up the treatment effect with
a power of 90%?

- Suppose that we would like to pick up an effect size of $\beta_1 = 60 g/kg$.
How many samples would be required in each group to obtain a power of 90%?

- Exercise: [Exercise7](./08_ExperimentalDesignII_1_rodents.html)
- Data path: Not required
<!-- - Solution: [Solution7](./08_ExperimentalDesignII_1_rodents_sol.html) -->

---

# Power analysis: the Puromycin dataset

Data on the velocity of an enzymatic reaction were obtained by Treloar (1974).
The number of counts per minute of radioactive product from the reaction was
measured as a function of substrate concentration in parts per million (ppm)
and from these counts the initial rate (or velocity) of the reaction was c
alculated (counts/min/min). The experiment was conducted once with
the enzyme treated with Puromycin, and once with the enzyme untreated.

We will answer four research questions:

- Use the data to calculate the power to pick up an association that is as
least as strong as the association you observed in the dataset when using
an experiment with the same design.

- Use the data to calculate the power to pick up an association where
the reaction rate increases on average with 10 counts/min when the substrate
concentration is 10 times higher ($\beta_1=10$).

- Use the data to calculate the number of repeats you need for each
concentration to pick up an association where the reaction rate increases
on average with 10 counts/min when the substrate concentration is 10 times
higher with a power of at least 90%. ($\beta_1=10$)

- Suppose that you would setup an experiment with a design similar with the
same concentrations as in the puromycin dataset and you have the following
restriction: you need to use each concentration at least once and can setup
at most 12 reactions, how would you choose your design points? Calculate
the power for this design when the effect size is 10 counts/min per 10 times
increase in the substrate concentration ($\beta_1=10$).

- Exercise: [Exercise8](./08_ExperimentalDesignII_1_puromycin.html)
- Data path: Not required
<!-- - Solution: [Solution8](./08_ExperimentalDesignII_1_puromycin_sol.html) -->

---


# Power analysis: he KPNA2 dataset

Histologic grade in breast cancer provides clinically important prognostic
information. Researchers examined whether histologic grade was associated with
gene expression profiles of breast cancers and whether such profiles could be
used to improve histologic grading.

In this tutorial we will assess the association between histologic grade and
the expression of the KPNA2 gene that is known to be associated with poor breast
cancer prognosis. The patients, however, do not only differ in the histologic
grade, but also on their lymph node status. The lymph nodes were
not affected (0) or surgically removed (1).

We will answer three research questions:

- What is the power to pick up each of the contrasts when their real effect
sizes would be equal to the effect sizes we observed in the study?

- How does the power evolves if we have 2 upto 10 repeats for each factor
combination of grade and node when their real effect sizes would be equal to
the ones we observed in the study?

- What is the power to pick up each of the contrasts when the FC for grade for patients with unaffected lymph nodes equals 1.5 ($\beta_g = log2(1.5)$)?

- Data path: "https://raw.githubusercontent.com/statOmics/SGA21/master/data/kpna2.txt"
<!-- - Solution: [Solution9](./08_ExperimentalDesignII_3_kpna2_sol.Rmd.html) -->
</div>
<div class="footer">
    <hr>
    This work is licensed under the <a href= "https://creativecommons.org/licenses/by-nc-sa/4.0">
    CC BY-NC-SA 4.0</a> licence.
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("08_exercises.Rmd");
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
